// @openrouter/sdk/esm/models/operations/createkeys.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e7a250c8b7bb
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
/**
 * Type of limit reset for the API key (daily, weekly, monthly, or null for no reset). Resets happen automatically at midnight UTC, and weeks are Monday through Sunday.
 */
export const CreateKeysLimitReset = {
    Daily: "daily",
    Weekly: "weekly",
    Monthly: "monthly",
};
/** @internal */
export const CreateKeysLimitReset$outboundSchema = openEnums.outboundSchema(CreateKeysLimitReset);
/** @internal */
export const CreateKeysRequest$outboundSchema = z.object({
    name: z.string(),
    limit: z.nullable(z.number()).optional(),
    limitReset: z.nullable(CreateKeysLimitReset$outboundSchema).optional(),
    includeByokInLimit: z.boolean().optional(),
    expiresAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
}).transform((v) => {
    return remap$(v, {
        limitReset: "limit_reset",
        includeByokInLimit: "include_byok_in_limit",
        expiresAt: "expires_at",
    });
});
export function createKeysRequestToJSON(createKeysRequest) {
    return JSON.stringify(CreateKeysRequest$outboundSchema.parse(createKeysRequest));
}
/** @internal */
export const CreateKeysData$inboundSchema = z.object({
    hash: z.string(),
    name: z.string(),
    label: z.string(),
    disabled: z.boolean(),
    limit: z.nullable(z.number()),
    limit_remaining: z.nullable(z.number()),
    limit_reset: z.nullable(z.string()),
    include_byok_in_limit: z.boolean(),
    usage: z.number(),
    usage_daily: z.number(),
    usage_weekly: z.number(),
    usage_monthly: z.number(),
    byok_usage: z.number(),
    byok_usage_daily: z.number(),
    byok_usage_weekly: z.number(),
    byok_usage_monthly: z.number(),
    created_at: z.string(),
    updated_at: z.nullable(z.string()),
    expires_at: z.nullable(z.iso.datetime({ offset: true }).transform(v => new Date(v))).optional(),
}).transform((v) => {
    return remap$(v, {
        "limit_remaining": "limitRemaining",
        "limit_reset": "limitReset",
        "include_byok_in_limit": "includeByokInLimit",
        "usage_daily": "usageDaily",
        "usage_weekly": "usageWeekly",
        "usage_monthly": "usageMonthly",
        "byok_usage": "byokUsage",
        "byok_usage_daily": "byokUsageDaily",
        "byok_usage_weekly": "byokUsageWeekly",
        "byok_usage_monthly": "byokUsageMonthly",
        "created_at": "createdAt",
        "updated_at": "updatedAt",
        "expires_at": "expiresAt",
    });
});
export function createKeysDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateKeysData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateKeysData' from JSON`);
}
/** @internal */
export const CreateKeysResponse$inboundSchema = z.object({
    data: z.lazy(() => CreateKeysData$inboundSchema),
    key: z.string(),
});
export function createKeysResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateKeysResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateKeysResponse' from JSON`);
}
