// @openrouter/sdk/esm/models/operations/createcoinbasecharge.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 1e75ac2debf2
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CreateCoinbaseChargeSecurity$outboundSchema = z.object({
    bearer: z.string(),
});
export function createCoinbaseChargeSecurityToJSON(createCoinbaseChargeSecurity) {
    return JSON.stringify(CreateCoinbaseChargeSecurity$outboundSchema.parse(createCoinbaseChargeSecurity));
}
/** @internal */
export const CallData$inboundSchema = z.object({
    deadline: z.string(),
    fee_amount: z.string(),
    id: z.string(),
    operator: z.string(),
    prefix: z.string(),
    recipient: z.string(),
    recipient_amount: z.string(),
    recipient_currency: z.string(),
    refund_destination: z.string(),
    signature: z.string(),
}).transform((v) => {
    return remap$(v, {
        "fee_amount": "feeAmount",
        "recipient_amount": "recipientAmount",
        "recipient_currency": "recipientCurrency",
        "refund_destination": "refundDestination",
    });
});
export function callDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CallData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CallData' from JSON`);
}
/** @internal */
export const Metadata$inboundSchema = z.object({
    chain_id: z.number(),
    contract_address: z.string(),
    sender: z.string(),
}).transform((v) => {
    return remap$(v, {
        "chain_id": "chainId",
        "contract_address": "contractAddress",
    });
});
export function metadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Metadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Metadata' from JSON`);
}
/** @internal */
export const TransferIntent$inboundSchema = z.object({
    call_data: z.lazy(() => CallData$inboundSchema),
    metadata: z.lazy(() => Metadata$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "call_data": "callData",
    });
});
export function transferIntentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TransferIntent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransferIntent' from JSON`);
}
/** @internal */
export const Web3Data$inboundSchema = z.object({
    transfer_intent: z.lazy(() => TransferIntent$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "transfer_intent": "transferIntent",
    });
});
export function web3DataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Web3Data$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Web3Data' from JSON`);
}
/** @internal */
export const CreateCoinbaseChargeData$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string(),
    expires_at: z.string(),
    web3_data: z.lazy(() => Web3Data$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "expires_at": "expiresAt",
        "web3_data": "web3Data",
    });
});
export function createCoinbaseChargeDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateCoinbaseChargeData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateCoinbaseChargeData' from JSON`);
}
/** @internal */
export const CreateCoinbaseChargeResponse$inboundSchema = z.object({
    data: z.lazy(() => CreateCoinbaseChargeData$inboundSchema),
});
export function createCoinbaseChargeResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateCoinbaseChargeResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateCoinbaseChargeResponse' from JSON`);
}
