// @openrouter/sdk/esm/models/operations/getkey.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 27d0f86c087f
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const GetKeyRequest$outboundSchema = z.object({
    hash: z.string(),
});
export function getKeyRequestToJSON(getKeyRequest) {
    return JSON.stringify(GetKeyRequest$outboundSchema.parse(getKeyRequest));
}
/** @internal */
export const GetKeyData$inboundSchema = z
    .object({
    hash: z.string(),
    name: z.string(),
    label: z.string(),
    disabled: z.boolean(),
    limit: z.nullable(z.number()),
    limit_remaining: z.nullable(z.number()),
    limit_reset: z.nullable(z.string()),
    include_byok_in_limit: z.boolean(),
    usage: z.number(),
    usage_daily: z.number(),
    usage_weekly: z.number(),
    usage_monthly: z.number(),
    byok_usage: z.number(),
    byok_usage_daily: z.number(),
    byok_usage_weekly: z.number(),
    byok_usage_monthly: z.number(),
    created_at: z.string(),
    updated_at: z.nullable(z.string()),
    expires_at: z.nullable(z.iso.datetime({ offset: true }).transform(v => new Date(v))).optional(),
}).transform((v) => {
    return remap$(v, {
        "limit_remaining": "limitRemaining",
        "limit_reset": "limitReset",
        "include_byok_in_limit": "includeByokInLimit",
        "usage_daily": "usageDaily",
        "usage_weekly": "usageWeekly",
        "usage_monthly": "usageMonthly",
        "byok_usage": "byokUsage",
        "byok_usage_daily": "byokUsageDaily",
        "byok_usage_weekly": "byokUsageWeekly",
        "byok_usage_monthly": "byokUsageMonthly",
        "created_at": "createdAt",
        "updated_at": "updatedAt",
        "expires_at": "expiresAt",
    });
});
export function getKeyDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetKeyData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetKeyData' from JSON`);
}
/** @internal */
export const GetKeyResponse$inboundSchema = z.object({
    data: z.lazy(() => GetKeyData$inboundSchema),
});
export function getKeyResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetKeyResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetKeyResponse' from JSON`);
}
