// @openrouter/sdk/esm/models/operations/getparameters.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 9b364a4cca61
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
import * as models from "../index.js";
export const SupportedParameter = {
    Temperature: "temperature",
    TopP: "top_p",
    TopK: "top_k",
    MinP: "min_p",
    TopA: "top_a",
    FrequencyPenalty: "frequency_penalty",
    PresencePenalty: "presence_penalty",
    RepetitionPenalty: "repetition_penalty",
    MaxTokens: "max_tokens",
    LogitBias: "logit_bias",
    Logprobs: "logprobs",
    TopLogprobs: "top_logprobs",
    Seed: "seed",
    ResponseFormat: "response_format",
    StructuredOutputs: "structured_outputs",
    Stop: "stop",
    Tools: "tools",
    ToolChoice: "tool_choice",
    ParallelToolCalls: "parallel_tool_calls",
    IncludeReasoning: "include_reasoning",
    Reasoning: "reasoning",
    ReasoningEffort: "reasoning_effort",
    WebSearchOptions: "web_search_options",
    Verbosity: "verbosity",
};
/** @internal */
export const GetParametersSecurity$outboundSchema = z.object({
    bearer: z.string(),
});
export function getParametersSecurityToJSON(getParametersSecurity) {
    return JSON.stringify(GetParametersSecurity$outboundSchema.parse(getParametersSecurity));
}
/** @internal */
export const GetParametersRequest$outboundSchema = z.object({
    author: z.string(),
    slug: z.string(),
    provider: models.ProviderName$outboundSchema.optional(),
});
export function getParametersRequestToJSON(getParametersRequest) {
    return JSON.stringify(GetParametersRequest$outboundSchema.parse(getParametersRequest));
}
/** @internal */
export const SupportedParameter$inboundSchema = openEnums.inboundSchema(SupportedParameter);
/** @internal */
export const GetParametersData$inboundSchema = z.object({
    model: z.string(),
    supported_parameters: z.array(SupportedParameter$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "supported_parameters": "supportedParameters",
    });
});
export function getParametersDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetParametersData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetParametersData' from JSON`);
}
/** @internal */
export const GetParametersResponse$inboundSchema = z.object({
    data: z.lazy(() => GetParametersData$inboundSchema),
});
export function getParametersResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetParametersResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetParametersResponse' from JSON`);
}
