// @openrouter/sdk/esm/models/operations/list.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ce11386ad30f
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const ListRequest$outboundSchema = z.object({
    includeDisabled: z.string().optional(),
    offset: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        includeDisabled: "include_disabled",
    });
});
export function listRequestToJSON(listRequest) {
    return JSON.stringify(ListRequest$outboundSchema.parse(listRequest));
}
/** @internal */
export const ListData$inboundSchema = z.object({
    hash: z.string(),
    name: z.string(),
    label: z.string(),
    disabled: z.boolean(),
    limit: z.nullable(z.number()),
    limit_remaining: z.nullable(z.number()),
    limit_reset: z.nullable(z.string()),
    include_byok_in_limit: z.boolean(),
    usage: z.number(),
    usage_daily: z.number(),
    usage_weekly: z.number(),
    usage_monthly: z.number(),
    byok_usage: z.number(),
    byok_usage_daily: z.number(),
    byok_usage_weekly: z.number(),
    byok_usage_monthly: z.number(),
    created_at: z.string(),
    updated_at: z.nullable(z.string()),
    expires_at: z.nullable(z.iso.datetime({ offset: true }).transform(v => new Date(v))).optional(),
}).transform((v) => {
    return remap$(v, {
        "limit_remaining": "limitRemaining",
        "limit_reset": "limitReset",
        "include_byok_in_limit": "includeByokInLimit",
        "usage_daily": "usageDaily",
        "usage_weekly": "usageWeekly",
        "usage_monthly": "usageMonthly",
        "byok_usage": "byokUsage",
        "byok_usage_daily": "byokUsageDaily",
        "byok_usage_weekly": "byokUsageWeekly",
        "byok_usage_monthly": "byokUsageMonthly",
        "created_at": "createdAt",
        "updated_at": "updatedAt",
        "expires_at": "expiresAt",
    });
});
export function listDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListData' from JSON`);
}
/** @internal */
export const ListResponse$inboundSchema = z
    .object({
    data: z.array(z.lazy(() => ListData$inboundSchema)),
});
export function listResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResponse' from JSON`);
}
