// @openrouter/sdk/esm/models/operations/listproviders.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 76a35169de06
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const ListProvidersData$inboundSchema = z.object({
    name: z.string(),
    slug: z.string(),
    privacy_policy_url: z.nullable(z.string()),
    terms_of_service_url: z.nullable(z.string()).optional(),
    status_page_url: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "privacy_policy_url": "privacyPolicyUrl",
        "terms_of_service_url": "termsOfServiceUrl",
        "status_page_url": "statusPageUrl",
    });
});
export function listProvidersDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListProvidersData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListProvidersData' from JSON`);
}
/** @internal */
export const ListProvidersResponse$inboundSchema = z.object({
    data: z.array(z.lazy(() => ListProvidersData$inboundSchema)),
});
export function listProvidersResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListProvidersResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListProvidersResponse' from JSON`);
}
