// @openrouter/sdk/esm/models/operations/updatekeys.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 30a0942aa2b9
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
/**
 * New limit reset type for the API key (daily, weekly, monthly, or null for no reset). Resets happen automatically at midnight UTC, and weeks are Monday through Sunday.
 */
export const UpdateKeysLimitReset = {
    Daily: "daily",
    Weekly: "weekly",
    Monthly: "monthly",
};
/** @internal */
export const UpdateKeysLimitReset$outboundSchema = openEnums.outboundSchema(UpdateKeysLimitReset);
/** @internal */
export const UpdateKeysRequestBody$outboundSchema = z.object({
    name: z.string().optional(),
    disabled: z.boolean().optional(),
    limit: z.nullable(z.number()).optional(),
    limitReset: z.nullable(UpdateKeysLimitReset$outboundSchema).optional(),
    includeByokInLimit: z.boolean().optional(),
}).transform((v) => {
    return remap$(v, {
        limitReset: "limit_reset",
        includeByokInLimit: "include_byok_in_limit",
    });
});
export function updateKeysRequestBodyToJSON(updateKeysRequestBody) {
    return JSON.stringify(UpdateKeysRequestBody$outboundSchema.parse(updateKeysRequestBody));
}
/** @internal */
export const UpdateKeysRequest$outboundSchema = z.object({
    hash: z.string(),
    requestBody: z.lazy(() => UpdateKeysRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function updateKeysRequestToJSON(updateKeysRequest) {
    return JSON.stringify(UpdateKeysRequest$outboundSchema.parse(updateKeysRequest));
}
/** @internal */
export const UpdateKeysData$inboundSchema = z.object({
    hash: z.string(),
    name: z.string(),
    label: z.string(),
    disabled: z.boolean(),
    limit: z.nullable(z.number()),
    limit_remaining: z.nullable(z.number()),
    limit_reset: z.nullable(z.string()),
    include_byok_in_limit: z.boolean(),
    usage: z.number(),
    usage_daily: z.number(),
    usage_weekly: z.number(),
    usage_monthly: z.number(),
    byok_usage: z.number(),
    byok_usage_daily: z.number(),
    byok_usage_weekly: z.number(),
    byok_usage_monthly: z.number(),
    created_at: z.string(),
    updated_at: z.nullable(z.string()),
    expires_at: z.nullable(z.iso.datetime({ offset: true }).transform(v => new Date(v))).optional(),
}).transform((v) => {
    return remap$(v, {
        "limit_remaining": "limitRemaining",
        "limit_reset": "limitReset",
        "include_byok_in_limit": "includeByokInLimit",
        "usage_daily": "usageDaily",
        "usage_weekly": "usageWeekly",
        "usage_monthly": "usageMonthly",
        "byok_usage": "byokUsage",
        "byok_usage_daily": "byokUsageDaily",
        "byok_usage_weekly": "byokUsageWeekly",
        "byok_usage_monthly": "byokUsageMonthly",
        "created_at": "createdAt",
        "updated_at": "updatedAt",
        "expires_at": "expiresAt",
    });
});
export function updateKeysDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateKeysData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateKeysData' from JSON`);
}
/** @internal */
export const UpdateKeysResponse$inboundSchema = z.object({
    data: z.lazy(() => UpdateKeysData$inboundSchema),
});
export function updateKeysResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateKeysResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateKeysResponse' from JSON`);
}
