// @openrouter/sdk/esm/sdk/apikeys.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5dda931501e1
 */
import { apiKeysCreate } from "../funcs/apiKeysCreate.js";
import { apiKeysDelete } from "../funcs/apiKeysDelete.js";
import { apiKeysGet } from "../funcs/apiKeysGet.js";
import { apiKeysGetCurrentKeyMetadata } from "../funcs/apiKeysGetCurrentKeyMetadata.js";
import { apiKeysList } from "../funcs/apiKeysList.js";
import { apiKeysUpdate } from "../funcs/apiKeysUpdate.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
export class APIKeys extends ClientSDK {
    /**
     * List API keys
     */
    async list(request, options) {
        return unwrapAsync(apiKeysList(this, request, options));
    }
    /**
     * Create a new API key
     */
    async create(request, options) {
        return unwrapAsync(apiKeysCreate(this, request, options));
    }
    /**
     * Update an API key
     */
    async update(request, options) {
        return unwrapAsync(apiKeysUpdate(this, request, options));
    }
    /**
     * Delete an API key
     */
    async delete(request, options) {
        return unwrapAsync(apiKeysDelete(this, request, options));
    }
    /**
     * Get a single API key
     */
    async get(request, options) {
        return unwrapAsync(apiKeysGet(this, request, options));
    }
    /**
     * Get current API key
     *
     * @remarks
     * Get information on the API key associated with the current authentication session
     */
    async getCurrentKeyMetadata(options) {
        return unwrapAsync(apiKeysGetCurrentKeyMetadata(this, options));
    }
}
