// @openrouter/sdk/esm/sdk/sdk.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 784571af2f69
 */
import { ClientSDK } from "../lib/sdks.js";
import { Analytics } from "./analytics.js";
import { APIKeys } from "./apikeys.js";
import { Beta } from "./beta.js";
import { Chat } from "./chat.js";
import { Completions } from "./completions.js";
import { Credits } from "./credits.js";
import { Embeddings } from "./embeddings.js";
import { Endpoints } from "./endpoints.js";
import { Generations } from "./generations.js";
import { Models } from "./models.js";
import { OAuth } from "./oauth.js";
import { ParametersT } from "./parameters.js";
import { Providers } from "./providers.js";
// #region imports
import { callModel as callModelFunc, } from "../funcs/call-model.js";
import { ToolType } from "../lib/tool-types.js";
export { ToolType };
// #endregion imports
export class OpenRouter extends ClientSDK {
    get beta() {
        return (this._beta ?? (this._beta = new Beta(this._options)));
    }
    get analytics() {
        return (this._analytics ?? (this._analytics = new Analytics(this._options)));
    }
    get credits() {
        return (this._credits ?? (this._credits = new Credits(this._options)));
    }
    get embeddings() {
        return (this._embeddings ?? (this._embeddings = new Embeddings(this._options)));
    }
    get generations() {
        return (this._generations ?? (this._generations = new Generations(this._options)));
    }
    get models() {
        return (this._models ?? (this._models = new Models(this._options)));
    }
    get endpoints() {
        return (this._endpoints ?? (this._endpoints = new Endpoints(this._options)));
    }
    get parameters() {
        return (this._parameters ?? (this._parameters = new ParametersT(this._options)));
    }
    get providers() {
        return (this._providers ?? (this._providers = new Providers(this._options)));
    }
    get apiKeys() {
        return (this._apiKeys ?? (this._apiKeys = new APIKeys(this._options)));
    }
    get oAuth() {
        return (this._oAuth ?? (this._oAuth = new OAuth(this._options)));
    }
    get chat() {
        return (this._chat ?? (this._chat = new Chat(this._options)));
    }
    get completions() {
        return (this._completions ?? (this._completions = new Completions(this._options)));
    }
    // #region sdk-class-body
    callModel(request, options) {
        return callModelFunc(this, request, options);
    }
}
