// @openrouter/sdk/esm/models/assistantmessage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 2147a953e992
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ChatMessageContentItem$inboundSchema, ChatMessageContentItem$outboundSchema, } from "./chatmessagecontentitem.js";
import { ChatMessageToolCall$inboundSchema, ChatMessageToolCall$outboundSchema, } from "./chatmessagetoolcall.js";
/** @internal */
export const AssistantMessageContent$inboundSchema = z.union([z.string(), z.array(ChatMessageContentItem$inboundSchema)]);
/** @internal */
export const AssistantMessageContent$outboundSchema = z.union([z.string(), z.array(ChatMessageContentItem$outboundSchema)]);
export function assistantMessageContentToJSON(assistantMessageContent) {
    return JSON.stringify(AssistantMessageContent$outboundSchema.parse(assistantMessageContent));
}
export function assistantMessageContentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AssistantMessageContent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AssistantMessageContent' from JSON`);
}
/** @internal */
export const AssistantMessage$inboundSchema = z.object({
    role: z.literal("assistant"),
    content: z.nullable(z.union([z.string(), z.array(ChatMessageContentItem$inboundSchema)])).optional(),
    name: z.string().optional(),
    tool_calls: z.array(ChatMessageToolCall$inboundSchema).optional(),
    refusal: z.nullable(z.string()).optional(),
    reasoning: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "tool_calls": "toolCalls",
    });
});
/** @internal */
export const AssistantMessage$outboundSchema = z.object({
    role: z.literal("assistant"),
    content: z.nullable(z.union([z.string(), z.array(ChatMessageContentItem$outboundSchema)])).optional(),
    name: z.string().optional(),
    toolCalls: z.array(ChatMessageToolCall$outboundSchema).optional(),
    refusal: z.nullable(z.string()).optional(),
    reasoning: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        toolCalls: "tool_calls",
    });
});
export function assistantMessageToJSON(assistantMessage) {
    return JSON.stringify(AssistantMessage$outboundSchema.parse(assistantMessage));
}
export function assistantMessageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AssistantMessage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AssistantMessage' from JSON`);
}
