// @openrouter/sdk/esm/models/chatmessagetoolcall.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: b294b44052da
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const ChatMessageToolCallFunction$inboundSchema = z.object({
    name: z.string(),
    arguments: z.string(),
});
/** @internal */
export const ChatMessageToolCallFunction$outboundSchema = z.object({
    name: z.string(),
    arguments: z.string(),
});
export function chatMessageToolCallFunctionToJSON(chatMessageToolCallFunction) {
    return JSON.stringify(ChatMessageToolCallFunction$outboundSchema.parse(chatMessageToolCallFunction));
}
export function chatMessageToolCallFunctionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageToolCallFunction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageToolCallFunction' from JSON`);
}
/** @internal */
export const ChatMessageToolCall$inboundSchema = z.object({
    id: z.string(),
    type: z.literal("function"),
    function: z.lazy(() => ChatMessageToolCallFunction$inboundSchema),
});
/** @internal */
export const ChatMessageToolCall$outboundSchema = z.object({
    id: z.string(),
    type: z.literal("function"),
    function: z.lazy(() => ChatMessageToolCallFunction$outboundSchema),
});
export function chatMessageToolCallToJSON(chatMessageToolCall) {
    return JSON.stringify(ChatMessageToolCall$outboundSchema.parse(chatMessageToolCall));
}
export function chatMessageToolCallFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageToolCall$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageToolCall' from JSON`);
}
