// @openrouter/sdk/esm/models/chatmessagecontentitemvideo.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5a81301cb4a7
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const VideoUrl2$inboundSchema = z.object({
    url: z.string(),
});
/** @internal */
export const VideoUrl2$outboundSchema = z.object({
    url: z.string(),
});
export function videoUrl2ToJSON(videoUrl2) {
    return JSON.stringify(VideoUrl2$outboundSchema.parse(videoUrl2));
}
export function videoUrl2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => VideoUrl2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'VideoUrl2' from JSON`);
}
/** @internal */
export const ChatMessageContentItemVideoVideoURL$inboundSchema = z.object({
    type: z.literal("video_url"),
    video_url: z.lazy(() => VideoUrl2$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "video_url": "videoUrl",
    });
});
/** @internal */
export const ChatMessageContentItemVideoVideoURL$outboundSchema = z.object({
    type: z.literal("video_url"),
    videoUrl: z.lazy(() => VideoUrl2$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        videoUrl: "video_url",
    });
});
export function chatMessageContentItemVideoVideoURLToJSON(chatMessageContentItemVideoVideoURL) {
    return JSON.stringify(ChatMessageContentItemVideoVideoURL$outboundSchema.parse(chatMessageContentItemVideoVideoURL));
}
export function chatMessageContentItemVideoVideoURLFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItemVideoVideoURL$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItemVideoVideoURL' from JSON`);
}
/** @internal */
export const VideoUrl1$inboundSchema = z.object({
    url: z.string(),
});
/** @internal */
export const VideoUrl1$outboundSchema = z.object({
    url: z.string(),
});
export function videoUrl1ToJSON(videoUrl1) {
    return JSON.stringify(VideoUrl1$outboundSchema.parse(videoUrl1));
}
export function videoUrl1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => VideoUrl1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'VideoUrl1' from JSON`);
}
/** @internal */
export const ChatMessageContentItemVideoInputVideo$inboundSchema = z.object({
    type: z.literal("input_video"),
    video_url: z.lazy(() => VideoUrl1$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "video_url": "videoUrl",
    });
});
/** @internal */
export const ChatMessageContentItemVideoInputVideo$outboundSchema = z.object({
    type: z.literal("input_video"),
    videoUrl: z.lazy(() => VideoUrl1$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        videoUrl: "video_url",
    });
});
export function chatMessageContentItemVideoInputVideoToJSON(chatMessageContentItemVideoInputVideo) {
    return JSON.stringify(ChatMessageContentItemVideoInputVideo$outboundSchema.parse(chatMessageContentItemVideoInputVideo));
}
export function chatMessageContentItemVideoInputVideoFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItemVideoInputVideo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItemVideoInputVideo' from JSON`);
}
/** @internal */
export const ChatMessageContentItemVideo$inboundSchema = z.union([
    z.lazy(() => ChatMessageContentItemVideoInputVideo$inboundSchema),
    z.lazy(() => ChatMessageContentItemVideoVideoURL$inboundSchema),
]);
/** @internal */
export const ChatMessageContentItemVideo$outboundSchema = z.union([
    z.lazy(() => ChatMessageContentItemVideoInputVideo$outboundSchema),
    z.lazy(() => ChatMessageContentItemVideoVideoURL$outboundSchema),
]);
export function chatMessageContentItemVideoToJSON(chatMessageContentItemVideo) {
    return JSON.stringify(ChatMessageContentItemVideo$outboundSchema.parse(chatMessageContentItemVideo));
}
export function chatMessageContentItemVideoFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItemVideo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItemVideo' from JSON`);
}
