// @openrouter/sdk/esm/models/chatmessagecontentitemimage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3989cdc41817
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const ChatMessageContentItemImageDetail = {
    Auto: "auto",
    Low: "low",
    High: "high",
};
/** @internal */
export const ChatMessageContentItemImageDetail$inboundSchema = openEnums.inboundSchema(ChatMessageContentItemImageDetail);
/** @internal */
export const ChatMessageContentItemImageDetail$outboundSchema = openEnums.outboundSchema(ChatMessageContentItemImageDetail);
/** @internal */
export const ImageUrl$inboundSchema = z.object({
    url: z.string(),
    detail: ChatMessageContentItemImageDetail$inboundSchema.optional(),
});
/** @internal */
export const ImageUrl$outboundSchema = z
    .object({
    url: z.string(),
    detail: ChatMessageContentItemImageDetail$outboundSchema.optional(),
});
export function imageUrlToJSON(imageUrl) {
    return JSON.stringify(ImageUrl$outboundSchema.parse(imageUrl));
}
export function imageUrlFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageUrl$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageUrl' from JSON`);
}
/** @internal */
export const ChatMessageContentItemImage$inboundSchema = z.object({
    type: z.literal("image_url"),
    image_url: z.lazy(() => ImageUrl$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "image_url": "imageUrl",
    });
});
/** @internal */
export const ChatMessageContentItemImage$outboundSchema = z.object({
    type: z.literal("image_url"),
    imageUrl: z.lazy(() => ImageUrl$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        imageUrl: "image_url",
    });
});
export function chatMessageContentItemImageToJSON(chatMessageContentItemImage) {
    return JSON.stringify(ChatMessageContentItemImage$outboundSchema.parse(chatMessageContentItemImage));
}
export function chatMessageContentItemImageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItemImage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItemImage' from JSON`);
}
