// @openrouter/sdk/esm/models/chatstreamingmessagetoolcall.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 74ba154581ba
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const ChatStreamingMessageToolCallFunction$inboundSchema = z.object({
    name: z.string().optional(),
    arguments: z.string().optional(),
});
export function chatStreamingMessageToolCallFunctionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatStreamingMessageToolCallFunction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatStreamingMessageToolCallFunction' from JSON`);
}
/** @internal */
export const ChatStreamingMessageToolCall$inboundSchema = z.object({
    index: z.number(),
    id: z.string().optional(),
    type: z.literal("function").optional(),
    function: z.lazy(() => ChatStreamingMessageToolCallFunction$inboundSchema)
        .optional(),
});
export function chatStreamingMessageToolCallFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatStreamingMessageToolCall$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatStreamingMessageToolCall' from JSON`);
}
