// @openrouter/sdk/esm/models/chatgenerationtokenusage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 1d57b0d238b8
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const CompletionTokensDetails$inboundSchema = z.object({
    reasoning_tokens: z.nullable(z.number()).optional(),
    audio_tokens: z.nullable(z.number()).optional(),
    accepted_prediction_tokens: z.nullable(z.number()).optional(),
    rejected_prediction_tokens: z.nullable(z.number()).optional(),
}).transform((v) => {
    return remap$(v, {
        "reasoning_tokens": "reasoningTokens",
        "audio_tokens": "audioTokens",
        "accepted_prediction_tokens": "acceptedPredictionTokens",
        "rejected_prediction_tokens": "rejectedPredictionTokens",
    });
});
export function completionTokensDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CompletionTokensDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CompletionTokensDetails' from JSON`);
}
/** @internal */
export const PromptTokensDetails$inboundSchema = z.object({
    cached_tokens: z.number().optional(),
    audio_tokens: z.number().optional(),
    video_tokens: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        "cached_tokens": "cachedTokens",
        "audio_tokens": "audioTokens",
        "video_tokens": "videoTokens",
    });
});
export function promptTokensDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PromptTokensDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PromptTokensDetails' from JSON`);
}
/** @internal */
export const ChatGenerationTokenUsage$inboundSchema = z.object({
    completion_tokens: z.number(),
    prompt_tokens: z.number(),
    total_tokens: z.number(),
    completion_tokens_details: z.nullable(z.lazy(() => CompletionTokensDetails$inboundSchema)).optional(),
    prompt_tokens_details: z.nullable(z.lazy(() => PromptTokensDetails$inboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        "completion_tokens": "completionTokens",
        "prompt_tokens": "promptTokens",
        "total_tokens": "totalTokens",
        "completion_tokens_details": "completionTokensDetails",
        "prompt_tokens_details": "promptTokensDetails",
    });
});
export function chatGenerationTokenUsageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatGenerationTokenUsage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatGenerationTokenUsage' from JSON`);
}
