// @openrouter/sdk/esm/models/chatmessagetokenlogprob.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: eed686a58eb1
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const TopLogprob$inboundSchema = z
    .object({
    token: z.string(),
    logprob: z.number(),
    bytes: z.nullable(z.array(z.number())),
});
export function topLogprobFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TopLogprob$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TopLogprob' from JSON`);
}
/** @internal */
export const ChatMessageTokenLogprob$inboundSchema = z.object({
    token: z.string(),
    logprob: z.number(),
    bytes: z.nullable(z.array(z.number())),
    top_logprobs: z.array(z.lazy(() => TopLogprob$inboundSchema)),
}).transform((v) => {
    return remap$(v, {
        "top_logprobs": "topLogprobs",
    });
});
export function chatMessageTokenLogprobFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageTokenLogprob$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageTokenLogprob' from JSON`);
}
