// @openrouter/sdk/esm/models/chatstreamingmessagechunk.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 339ee7e4b920
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ChatStreamingMessageToolCall$inboundSchema, } from "./chatstreamingmessagetoolcall.js";
import { Schema3$inboundSchema } from "./schema3.js";
export const ChatStreamingMessageChunkRole = {
    Assistant: "assistant",
};
/** @internal */
export const ChatStreamingMessageChunkRole$inboundSchema = z.enum(ChatStreamingMessageChunkRole);
/** @internal */
export const ChatStreamingMessageChunk$inboundSchema = z.object({
    role: ChatStreamingMessageChunkRole$inboundSchema.optional(),
    content: z.nullable(z.string()).optional(),
    reasoning: z.nullable(z.string()).optional(),
    refusal: z.nullable(z.string()).optional(),
    tool_calls: z.array(ChatStreamingMessageToolCall$inboundSchema).optional(),
    reasoning_details: z.array(Schema3$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "tool_calls": "toolCalls",
        "reasoning_details": "reasoningDetails",
    });
});
export function chatStreamingMessageChunkFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatStreamingMessageChunk$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatStreamingMessageChunk' from JSON`);
}
