// @openrouter/sdk/esm/models/chatmessagecontentitemtext.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 27089860fd98
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ChatMessageContentItemCacheControl$inboundSchema, ChatMessageContentItemCacheControl$outboundSchema, } from "./chatmessagecontentitemcachecontrol.js";
/** @internal */
export const ChatMessageContentItemText$inboundSchema = z.object({
    type: z.literal("text"),
    text: z.string(),
    cache_control: ChatMessageContentItemCacheControl$inboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        "cache_control": "cacheControl",
    });
});
/** @internal */
export const ChatMessageContentItemText$outboundSchema = z.object({
    type: z.literal("text"),
    text: z.string(),
    cacheControl: ChatMessageContentItemCacheControl$outboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
    });
});
export function chatMessageContentItemTextToJSON(chatMessageContentItemText) {
    return JSON.stringify(ChatMessageContentItemText$outboundSchema.parse(chatMessageContentItemText));
}
export function chatMessageContentItemTextFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItemText$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItemText' from JSON`);
}
