// @openrouter/sdk/esm/models/chatstreamingresponsechunk.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 00bf2ea52439
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ChatGenerationTokenUsage$inboundSchema, } from "./chatgenerationtokenusage.js";
import { ChatStreamingChoice$inboundSchema, } from "./chatstreamingchoice.js";
/** @internal */
export const ChatStreamingResponseChunkError$inboundSchema = z.object({
    message: z.string(),
    code: z.number(),
});
export function chatStreamingResponseChunkErrorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatStreamingResponseChunkError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatStreamingResponseChunkError' from JSON`);
}
/** @internal */
export const ChatStreamingResponseChunkData$inboundSchema = z.object({
    id: z.string(),
    choices: z.array(ChatStreamingChoice$inboundSchema),
    created: z.number(),
    model: z.string(),
    object: z.literal("chat.completion.chunk"),
    system_fingerprint: z.nullable(z.string()).optional(),
    error: z.lazy(() => ChatStreamingResponseChunkError$inboundSchema).optional(),
    usage: ChatGenerationTokenUsage$inboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        "system_fingerprint": "systemFingerprint",
    });
});
export function chatStreamingResponseChunkDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatStreamingResponseChunkData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatStreamingResponseChunkData' from JSON`);
}
/** @internal */
export const ChatStreamingResponseChunk$inboundSchema = z.object({
    data: z.string().transform((v, ctx) => {
        try {
            return JSON.parse(v);
        }
        catch (err) {
            ctx.addIssue({
                input: v,
                code: "custom",
                message: `malformed json: ${err}`,
            });
            return z.NEVER;
        }
    }).pipe(z.lazy(() => ChatStreamingResponseChunkData$inboundSchema)),
});
export function chatStreamingResponseChunkFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatStreamingResponseChunk$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatStreamingResponseChunk' from JSON`);
}
