// @openrouter/sdk/esm/models/chatmessagecontentitemcachecontrol.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: b5c18e04e19c
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const Ttl = {
    Fivem: "5m",
    Oneh: "1h",
};
/** @internal */
export const Ttl$inboundSchema = openEnums
    .inboundSchema(Ttl);
/** @internal */
export const Ttl$outboundSchema = openEnums
    .outboundSchema(Ttl);
/** @internal */
export const ChatMessageContentItemCacheControl$inboundSchema = z.object({
    type: z.literal("ephemeral"),
    ttl: Ttl$inboundSchema.optional(),
});
/** @internal */
export const ChatMessageContentItemCacheControl$outboundSchema = z.object({
    type: z.literal("ephemeral"),
    ttl: Ttl$outboundSchema.optional(),
});
export function chatMessageContentItemCacheControlToJSON(chatMessageContentItemCacheControl) {
    return JSON.stringify(ChatMessageContentItemCacheControl$outboundSchema.parse(chatMessageContentItemCacheControl));
}
export function chatMessageContentItemCacheControlFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItemCacheControl$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItemCacheControl' from JSON`);
}
