// @openrouter/sdk/esm/models/chatstreamingchoice.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 15346f0b1bc4
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ChatCompletionFinishReason$inboundSchema, } from "./chatcompletionfinishreason.js";
import { ChatMessageTokenLogprobs$inboundSchema, } from "./chatmessagetokenlogprobs.js";
import { ChatStreamingMessageChunk$inboundSchema, } from "./chatstreamingmessagechunk.js";
/** @internal */
export const ChatStreamingChoice$inboundSchema = z.object({
    delta: ChatStreamingMessageChunk$inboundSchema,
    finish_reason: z.nullable(ChatCompletionFinishReason$inboundSchema),
    index: z.number(),
    logprobs: z.nullable(ChatMessageTokenLogprobs$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "finish_reason": "finishReason",
    });
});
export function chatStreamingChoiceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatStreamingChoice$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatStreamingChoice' from JSON`);
}
