// @openrouter/sdk/esm/models/chatresponsechoice.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7db13c3cc866
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { AssistantMessage$inboundSchema, } from "./assistantmessage.js";
import { ChatCompletionFinishReason$inboundSchema, } from "./chatcompletionfinishreason.js";
import { ChatMessageTokenLogprobs$inboundSchema, } from "./chatmessagetokenlogprobs.js";
import { Schema3$inboundSchema } from "./schema3.js";
/** @internal */
export const ChatResponseChoice$inboundSchema = z.object({
    finish_reason: z.nullable(ChatCompletionFinishReason$inboundSchema),
    index: z.number(),
    message: AssistantMessage$inboundSchema,
    reasoning_details: z.array(Schema3$inboundSchema).optional(),
    logprobs: z.nullable(ChatMessageTokenLogprobs$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "finish_reason": "finishReason",
        "reasoning_details": "reasoningDetails",
    });
});
export function chatResponseChoiceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatResponseChoice$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatResponseChoice' from JSON`);
}
