// @openrouter/sdk/esm/models/completionresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 316ab9f7e558
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { CompletionChoice$inboundSchema, } from "./completionchoice.js";
import { CompletionUsage$inboundSchema, } from "./completionusage.js";
/** @internal */
export const CompletionResponse$inboundSchema = z.object({
    id: z.string(),
    object: z.literal("text_completion"),
    created: z.number(),
    model: z.string(),
    provider: z.string().optional(),
    system_fingerprint: z.string().optional(),
    choices: z.array(CompletionChoice$inboundSchema),
    usage: CompletionUsage$inboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        "system_fingerprint": "systemFingerprint",
    });
});
export function completionResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CompletionResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CompletionResponse' from JSON`);
}
