// @openrouter/sdk/esm/models/completionchoice.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 86092a0ff4c4
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { CompletionLogprobs$inboundSchema, } from "./completionlogprobs.js";
export const CompletionFinishReason = {
    Stop: "stop",
    Length: "length",
    ContentFilter: "content_filter",
};
/** @internal */
export const CompletionFinishReason$inboundSchema = openEnums.inboundSchema(CompletionFinishReason);
/** @internal */
export const CompletionChoice$inboundSchema = z.object({
    text: z.string(),
    index: z.number(),
    logprobs: z.nullable(CompletionLogprobs$inboundSchema),
    finish_reason: z.nullable(CompletionFinishReason$inboundSchema),
    native_finish_reason: z.string().optional(),
    reasoning: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "finish_reason": "finishReason",
        "native_finish_reason": "nativeFinishReason",
    });
});
export function completionChoiceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CompletionChoice$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CompletionChoice' from JSON`);
}
