// @openrouter/sdk/esm/models/completioncreateparams.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 44cc0c9c4c06
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { ResponseFormatJSONSchema$outboundSchema, } from "./responseformatjsonschema.js";
import { ResponseFormatTextGrammar$outboundSchema, } from "./responseformattextgrammar.js";
/** @internal */
export const Prompt$outboundSchema = z
    .union([
    z.string(),
    z.array(z.string()),
    z.array(z.number()),
    z.array(z.array(z.number())),
]);
export function promptToJSON(prompt) {
    return JSON.stringify(Prompt$outboundSchema.parse(prompt));
}
/** @internal */
export const CompletionCreateParamsStop$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function completionCreateParamsStopToJSON(completionCreateParamsStop) {
    return JSON.stringify(CompletionCreateParamsStop$outboundSchema.parse(completionCreateParamsStop));
}
/** @internal */
export const StreamOptions$outboundSchema = z.object({
    includeUsage: z.nullable(z.boolean()).optional(),
}).transform((v) => {
    return remap$(v, {
        includeUsage: "include_usage",
    });
});
export function streamOptionsToJSON(streamOptions) {
    return JSON.stringify(StreamOptions$outboundSchema.parse(streamOptions));
}
/** @internal */
export const CompletionCreateParamsResponseFormatPython$outboundSchema = z.object({
    type: z.literal("python"),
});
export function completionCreateParamsResponseFormatPythonToJSON(completionCreateParamsResponseFormatPython) {
    return JSON.stringify(CompletionCreateParamsResponseFormatPython$outboundSchema.parse(completionCreateParamsResponseFormatPython));
}
/** @internal */
export const CompletionCreateParamsResponseFormatJSONObject$outboundSchema = z.object({
    type: z.literal("json_object"),
});
export function completionCreateParamsResponseFormatJSONObjectToJSON(completionCreateParamsResponseFormatJSONObject) {
    return JSON.stringify(CompletionCreateParamsResponseFormatJSONObject$outboundSchema.parse(completionCreateParamsResponseFormatJSONObject));
}
/** @internal */
export const CompletionCreateParamsResponseFormatText$outboundSchema = z.object({
    type: z.literal("text"),
});
export function completionCreateParamsResponseFormatTextToJSON(completionCreateParamsResponseFormatText) {
    return JSON.stringify(CompletionCreateParamsResponseFormatText$outboundSchema.parse(completionCreateParamsResponseFormatText));
}
/** @internal */
export const CompletionCreateParamsResponseFormatUnion$outboundSchema = z.union([
    z.lazy(() => CompletionCreateParamsResponseFormatText$outboundSchema),
    z.lazy(() => CompletionCreateParamsResponseFormatJSONObject$outboundSchema),
    ResponseFormatJSONSchema$outboundSchema,
    ResponseFormatTextGrammar$outboundSchema,
    z.lazy(() => CompletionCreateParamsResponseFormatPython$outboundSchema),
]);
export function completionCreateParamsResponseFormatUnionToJSON(completionCreateParamsResponseFormatUnion) {
    return JSON.stringify(CompletionCreateParamsResponseFormatUnion$outboundSchema.parse(completionCreateParamsResponseFormatUnion));
}
/** @internal */
export const CompletionCreateParams$outboundSchema = z.object({
    model: z.string().optional(),
    models: z.array(z.string()).optional(),
    prompt: z.union([
        z.string(),
        z.array(z.string()),
        z.array(z.number()),
        z.array(z.array(z.number())),
    ]),
    bestOf: z.nullable(z.int()).optional(),
    echo: z.nullable(z.boolean()).optional(),
    frequencyPenalty: z.nullable(z.number()).optional(),
    logitBias: z.nullable(z.record(z.string(), z.number())).optional(),
    logprobs: z.nullable(z.int()).optional(),
    maxTokens: z.nullable(z.int()).optional(),
    n: z.nullable(z.int()).optional(),
    presencePenalty: z.nullable(z.number()).optional(),
    seed: z.nullable(z.int()).optional(),
    stop: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    stream: z.boolean().default(false),
    streamOptions: z.nullable(z.lazy(() => StreamOptions$outboundSchema))
        .optional(),
    suffix: z.nullable(z.string()).optional(),
    temperature: z.nullable(z.number()).optional(),
    topP: z.nullable(z.number()).optional(),
    user: z.string().optional(),
    metadata: z.nullable(z.record(z.string(), z.string())).optional(),
    responseFormat: z.nullable(z.union([
        z.lazy(() => CompletionCreateParamsResponseFormatText$outboundSchema),
        z.lazy(() => CompletionCreateParamsResponseFormatJSONObject$outboundSchema),
        ResponseFormatJSONSchema$outboundSchema,
        ResponseFormatTextGrammar$outboundSchema,
        z.lazy(() => CompletionCreateParamsResponseFormatPython$outboundSchema),
    ])).optional(),
}).transform((v) => {
    return remap$(v, {
        bestOf: "best_of",
        frequencyPenalty: "frequency_penalty",
        logitBias: "logit_bias",
        maxTokens: "max_tokens",
        presencePenalty: "presence_penalty",
        streamOptions: "stream_options",
        topP: "top_p",
        responseFormat: "response_format",
    });
});
export function completionCreateParamsToJSON(completionCreateParams) {
    return JSON.stringify(CompletionCreateParams$outboundSchema.parse(completionCreateParams));
}
