// @openrouter/sdk/esm/models/listendpointsresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5178e92a44ba
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { InputModality$inboundSchema } from "./inputmodality.js";
import { InstructType$inboundSchema } from "./instructtype.js";
import { OutputModality$inboundSchema, } from "./outputmodality.js";
import { PublicEndpoint$inboundSchema, } from "./publicendpoint.js";
/**
 * Tokenizer type used by the model
 */
export const Tokenizer = {
    Router: "Router",
    Media: "Media",
    Other: "Other",
    Gpt: "GPT",
    Claude: "Claude",
    Gemini: "Gemini",
    Grok: "Grok",
    Cohere: "Cohere",
    Nova: "Nova",
    Qwen: "Qwen",
    Yi: "Yi",
    DeepSeek: "DeepSeek",
    Mistral: "Mistral",
    Llama2: "Llama2",
    Llama3: "Llama3",
    Llama4: "Llama4",
    PaLM: "PaLM",
    Rwkv: "RWKV",
    Qwen3: "Qwen3",
};
/** @internal */
export const Tokenizer$inboundSchema = openEnums
    .inboundSchema(Tokenizer);
/** @internal */
export const Architecture$inboundSchema = z
    .object({
    tokenizer: z.nullable(Tokenizer$inboundSchema),
    instruct_type: z.nullable(InstructType$inboundSchema),
    modality: z.nullable(z.string()),
    input_modalities: z.array(InputModality$inboundSchema),
    output_modalities: z.array(OutputModality$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "instruct_type": "instructType",
        "input_modalities": "inputModalities",
        "output_modalities": "outputModalities",
    });
});
export function architectureFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Architecture$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Architecture' from JSON`);
}
/** @internal */
export const ListEndpointsResponse$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    created: z.number(),
    description: z.string(),
    architecture: z.lazy(() => Architecture$inboundSchema),
    endpoints: z.array(PublicEndpoint$inboundSchema),
});
export function listEndpointsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListEndpointsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListEndpointsResponse' from JSON`);
}
