// @openrouter/sdk/esm/models/message.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 34645e53d993
 */
import * as z from "zod/v4";
import { AssistantMessage$outboundSchema, } from "./assistantmessage.js";
import { ChatMessageContentItemText$outboundSchema, } from "./chatmessagecontentitemtext.js";
import { SystemMessage$outboundSchema, } from "./systemmessage.js";
import { ToolResponseMessage$outboundSchema, } from "./toolresponsemessage.js";
import { UserMessage$outboundSchema, } from "./usermessage.js";
/** @internal */
export const MessageContent$outboundSchema = z.union([z.string(), z.array(ChatMessageContentItemText$outboundSchema)]);
export function messageContentToJSON(messageContent) {
    return JSON.stringify(MessageContent$outboundSchema.parse(messageContent));
}
/** @internal */
export const MessageDeveloper$outboundSchema = z.object({
    role: z.literal("developer"),
    content: z.union([
        z.string(),
        z.array(ChatMessageContentItemText$outboundSchema),
    ]),
    name: z.string().optional(),
});
export function messageDeveloperToJSON(messageDeveloper) {
    return JSON.stringify(MessageDeveloper$outboundSchema.parse(messageDeveloper));
}
/** @internal */
export const Message$outboundSchema = z
    .union([
    SystemMessage$outboundSchema,
    UserMessage$outboundSchema,
    z.lazy(() => MessageDeveloper$outboundSchema),
    AssistantMessage$outboundSchema,
    ToolResponseMessage$outboundSchema,
]);
export function messageToJSON(message) {
    return JSON.stringify(Message$outboundSchema.parse(message));
}
