// @openrouter/sdk/esm/models/openresponseseasyinputmessage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 9cae543ceb47
 */
import * as z from "zod/v4";
import { ResponseInputAudio$outboundSchema, } from "./responseinputaudio.js";
import { ResponseInputFile$outboundSchema, } from "./responseinputfile.js";
import { ResponseInputImage$outboundSchema, } from "./responseinputimage.js";
import { ResponseInputText$outboundSchema, } from "./responseinputtext.js";
export const OpenResponsesEasyInputMessageType = {
    Message: "message",
};
export const OpenResponsesEasyInputMessageRoleDeveloper = {
    Developer: "developer",
};
export const OpenResponsesEasyInputMessageRoleAssistant = {
    Assistant: "assistant",
};
export const OpenResponsesEasyInputMessageRoleSystem = {
    System: "system",
};
export const OpenResponsesEasyInputMessageRoleUser = {
    User: "user",
};
/** @internal */
export const OpenResponsesEasyInputMessageType$outboundSchema = z.enum(OpenResponsesEasyInputMessageType);
/** @internal */
export const OpenResponsesEasyInputMessageRoleDeveloper$outboundSchema = z.enum(OpenResponsesEasyInputMessageRoleDeveloper);
/** @internal */
export const OpenResponsesEasyInputMessageRoleAssistant$outboundSchema = z.enum(OpenResponsesEasyInputMessageRoleAssistant);
/** @internal */
export const OpenResponsesEasyInputMessageRoleSystem$outboundSchema = z.enum(OpenResponsesEasyInputMessageRoleSystem);
/** @internal */
export const OpenResponsesEasyInputMessageRoleUser$outboundSchema = z.enum(OpenResponsesEasyInputMessageRoleUser);
/** @internal */
export const OpenResponsesEasyInputMessageRoleUnion$outboundSchema = z.union([
    OpenResponsesEasyInputMessageRoleUser$outboundSchema,
    OpenResponsesEasyInputMessageRoleSystem$outboundSchema,
    OpenResponsesEasyInputMessageRoleAssistant$outboundSchema,
    OpenResponsesEasyInputMessageRoleDeveloper$outboundSchema,
]);
export function openResponsesEasyInputMessageRoleUnionToJSON(openResponsesEasyInputMessageRoleUnion) {
    return JSON.stringify(OpenResponsesEasyInputMessageRoleUnion$outboundSchema.parse(openResponsesEasyInputMessageRoleUnion));
}
/** @internal */
export const OpenResponsesEasyInputMessageContent1$outboundSchema = z.union([
    ResponseInputText$outboundSchema,
    ResponseInputImage$outboundSchema,
    ResponseInputFile$outboundSchema,
    ResponseInputAudio$outboundSchema,
]);
export function openResponsesEasyInputMessageContent1ToJSON(openResponsesEasyInputMessageContent1) {
    return JSON.stringify(OpenResponsesEasyInputMessageContent1$outboundSchema.parse(openResponsesEasyInputMessageContent1));
}
/** @internal */
export const OpenResponsesEasyInputMessageContent2$outboundSchema = z.union([
    z.array(z.union([
        ResponseInputText$outboundSchema,
        ResponseInputImage$outboundSchema,
        ResponseInputFile$outboundSchema,
        ResponseInputAudio$outboundSchema,
    ])),
    z.string(),
]);
export function openResponsesEasyInputMessageContent2ToJSON(openResponsesEasyInputMessageContent2) {
    return JSON.stringify(OpenResponsesEasyInputMessageContent2$outboundSchema.parse(openResponsesEasyInputMessageContent2));
}
/** @internal */
export const OpenResponsesEasyInputMessage$outboundSchema = z.object({
    type: OpenResponsesEasyInputMessageType$outboundSchema.optional(),
    role: z.union([
        OpenResponsesEasyInputMessageRoleUser$outboundSchema,
        OpenResponsesEasyInputMessageRoleSystem$outboundSchema,
        OpenResponsesEasyInputMessageRoleAssistant$outboundSchema,
        OpenResponsesEasyInputMessageRoleDeveloper$outboundSchema,
    ]),
    content: z.union([
        z.array(z.union([
            ResponseInputText$outboundSchema,
            ResponseInputImage$outboundSchema,
            ResponseInputFile$outboundSchema,
            ResponseInputAudio$outboundSchema,
        ])),
        z.string(),
    ]),
});
export function openResponsesEasyInputMessageToJSON(openResponsesEasyInputMessage) {
    return JSON.stringify(OpenResponsesEasyInputMessage$outboundSchema.parse(openResponsesEasyInputMessage));
}
