// @openrouter/sdk/esm/models/openresponsesfunctiontoolcall.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: b028e3118a4e
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OpenResponsesFunctionToolCallType = {
    FunctionCall: "function_call",
};
/** @internal */
export const OpenResponsesFunctionToolCallType$outboundSchema = z.enum(OpenResponsesFunctionToolCallType);
/** @internal */
export const OpenResponsesFunctionToolCall$outboundSchema = z.object({
    type: OpenResponsesFunctionToolCallType$outboundSchema,
    callId: z.string(),
    name: z.string(),
    arguments: z.string(),
    id: z.string(),
    status: z.nullable(ToolCallStatus$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        callId: "call_id",
    });
});
export function openResponsesFunctionToolCallToJSON(openResponsesFunctionToolCall) {
    return JSON.stringify(OpenResponsesFunctionToolCall$outboundSchema.parse(openResponsesFunctionToolCall));
}
