// @openrouter/sdk/esm/models/openresponsesreasoning.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 296442f2d24a
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { ReasoningSummaryText$outboundSchema, } from "./reasoningsummarytext.js";
import { ReasoningTextContent$outboundSchema, } from "./reasoningtextcontent.js";
export const OpenResponsesReasoningType = {
    Reasoning: "reasoning",
};
export const OpenResponsesReasoningStatusInProgress = {
    InProgress: "in_progress",
};
export const OpenResponsesReasoningStatusIncomplete = {
    Incomplete: "incomplete",
};
export const OpenResponsesReasoningStatusCompleted = {
    Completed: "completed",
};
export const OpenResponsesReasoningFormat = {
    Unknown: "unknown",
    OpenaiResponsesV1: "openai-responses-v1",
    XaiResponsesV1: "xai-responses-v1",
    AnthropicClaudeV1: "anthropic-claude-v1",
    GoogleGeminiV1: "google-gemini-v1",
};
/** @internal */
export const OpenResponsesReasoningType$outboundSchema = z.enum(OpenResponsesReasoningType);
/** @internal */
export const OpenResponsesReasoningStatusInProgress$outboundSchema = z.enum(OpenResponsesReasoningStatusInProgress);
/** @internal */
export const OpenResponsesReasoningStatusIncomplete$outboundSchema = z.enum(OpenResponsesReasoningStatusIncomplete);
/** @internal */
export const OpenResponsesReasoningStatusCompleted$outboundSchema = z.enum(OpenResponsesReasoningStatusCompleted);
/** @internal */
export const OpenResponsesReasoningStatusUnion$outboundSchema = z.union([
    OpenResponsesReasoningStatusCompleted$outboundSchema,
    OpenResponsesReasoningStatusIncomplete$outboundSchema,
    OpenResponsesReasoningStatusInProgress$outboundSchema,
]);
export function openResponsesReasoningStatusUnionToJSON(openResponsesReasoningStatusUnion) {
    return JSON.stringify(OpenResponsesReasoningStatusUnion$outboundSchema.parse(openResponsesReasoningStatusUnion));
}
/** @internal */
export const OpenResponsesReasoningFormat$outboundSchema = openEnums.outboundSchema(OpenResponsesReasoningFormat);
/** @internal */
export const OpenResponsesReasoning$outboundSchema = z.object({
    type: OpenResponsesReasoningType$outboundSchema,
    id: z.string(),
    content: z.array(ReasoningTextContent$outboundSchema).optional(),
    summary: z.array(ReasoningSummaryText$outboundSchema),
    encryptedContent: z.nullable(z.string()).optional(),
    status: z.union([
        OpenResponsesReasoningStatusCompleted$outboundSchema,
        OpenResponsesReasoningStatusIncomplete$outboundSchema,
        OpenResponsesReasoningStatusInProgress$outboundSchema,
    ]).optional(),
    signature: z.nullable(z.string()).optional(),
    format: z.nullable(OpenResponsesReasoningFormat$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        encryptedContent: "encrypted_content",
    });
});
export function openResponsesReasoningToJSON(openResponsesReasoning) {
    return JSON.stringify(OpenResponsesReasoning$outboundSchema.parse(openResponsesReasoning));
}
