// @openrouter/sdk/esm/models/openresponsesstreamevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: bdd51c7b4bd0
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { OpenAIResponsesAnnotation$inboundSchema, } from "./openairesponsesannotation.js";
import { OpenAIResponsesRefusalContent$inboundSchema, } from "./openairesponsesrefusalcontent.js";
import { OpenResponsesErrorEvent$inboundSchema, } from "./openresponseserrorevent.js";
import { OpenResponsesImageGenCallCompleted$inboundSchema, } from "./openresponsesimagegencallcompleted.js";
import { OpenResponsesImageGenCallGenerating$inboundSchema, } from "./openresponsesimagegencallgenerating.js";
import { OpenResponsesImageGenCallInProgress$inboundSchema, } from "./openresponsesimagegencallinprogress.js";
import { OpenResponsesImageGenCallPartialImage$inboundSchema, } from "./openresponsesimagegencallpartialimage.js";
import { OpenResponsesLogProbs$inboundSchema, } from "./openresponseslogprobs.js";
import { OpenResponsesNonStreamingResponse$inboundSchema, } from "./openresponsesnonstreamingresponse.js";
import { OpenResponsesReasoningDeltaEvent$inboundSchema, } from "./openresponsesreasoningdeltaevent.js";
import { OpenResponsesReasoningDoneEvent$inboundSchema, } from "./openresponsesreasoningdoneevent.js";
import { OpenResponsesReasoningSummaryPartAddedEvent$inboundSchema, } from "./openresponsesreasoningsummarypartaddedevent.js";
import { OpenResponsesReasoningSummaryTextDeltaEvent$inboundSchema, } from "./openresponsesreasoningsummarytextdeltaevent.js";
import { OpenResponsesReasoningSummaryTextDoneEvent$inboundSchema, } from "./openresponsesreasoningsummarytextdoneevent.js";
import { ReasoningSummaryText$inboundSchema, } from "./reasoningsummarytext.js";
import { ReasoningTextContent$inboundSchema, } from "./reasoningtextcontent.js";
import { ResponseOutputText$inboundSchema, } from "./responseoutputtext.js";
import { ResponsesOutputItem$inboundSchema, } from "./responsesoutputitem.js";
/** @internal */
export const OpenResponsesStreamEventResponseReasoningSummaryPartDone$inboundSchema = z.object({
    type: z.literal("response.reasoning_summary_part.done"),
    output_index: z.number(),
    item_id: z.string(),
    summary_index: z.number(),
    part: ReasoningSummaryText$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "item_id": "itemId",
        "summary_index": "summaryIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseReasoningSummaryPartDoneFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseReasoningSummaryPartDone$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseReasoningSummaryPartDone' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseFunctionCallArgumentsDone$inboundSchema = z.object({
    type: z.literal("response.function_call_arguments.done"),
    item_id: z.string(),
    output_index: z.number(),
    name: z.string(),
    arguments: z.string(),
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "item_id": "itemId",
        "output_index": "outputIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseFunctionCallArgumentsDoneFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseFunctionCallArgumentsDone$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseFunctionCallArgumentsDone' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseFunctionCallArgumentsDelta$inboundSchema = z.object({
    type: z.literal("response.function_call_arguments.delta"),
    item_id: z.string(),
    output_index: z.number(),
    delta: z.string(),
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "item_id": "itemId",
        "output_index": "outputIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseFunctionCallArgumentsDeltaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseFunctionCallArgumentsDelta$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseFunctionCallArgumentsDelta' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseOutputTextAnnotationAdded$inboundSchema = z.object({
    type: z.literal("response.output_text.annotation.added"),
    output_index: z.number(),
    item_id: z.string(),
    content_index: z.number(),
    sequence_number: z.number(),
    annotation_index: z.number(),
    annotation: OpenAIResponsesAnnotation$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "item_id": "itemId",
        "content_index": "contentIndex",
        "sequence_number": "sequenceNumber",
        "annotation_index": "annotationIndex",
    });
});
export function openResponsesStreamEventResponseOutputTextAnnotationAddedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseOutputTextAnnotationAdded$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseOutputTextAnnotationAdded' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseRefusalDone$inboundSchema = z.object({
    type: z.literal("response.refusal.done"),
    output_index: z.number(),
    item_id: z.string(),
    content_index: z.number(),
    refusal: z.string(),
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "item_id": "itemId",
        "content_index": "contentIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseRefusalDoneFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseRefusalDone$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseRefusalDone' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseRefusalDelta$inboundSchema = z.object({
    type: z.literal("response.refusal.delta"),
    output_index: z.number(),
    item_id: z.string(),
    content_index: z.number(),
    delta: z.string(),
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "item_id": "itemId",
        "content_index": "contentIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseRefusalDeltaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseRefusalDelta$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseRefusalDelta' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseOutputTextDone$inboundSchema = z.object({
    type: z.literal("response.output_text.done"),
    output_index: z.number(),
    item_id: z.string(),
    content_index: z.number(),
    text: z.string(),
    sequence_number: z.number(),
    logprobs: z.array(OpenResponsesLogProbs$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "item_id": "itemId",
        "content_index": "contentIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseOutputTextDoneFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseOutputTextDone$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseOutputTextDone' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseOutputTextDelta$inboundSchema = z
    .object({
    type: z.literal("response.output_text.delta"),
    logprobs: z.array(OpenResponsesLogProbs$inboundSchema),
    output_index: z.number(),
    item_id: z.string(),
    content_index: z.number(),
    delta: z.string(),
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "item_id": "itemId",
        "content_index": "contentIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseOutputTextDeltaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseOutputTextDelta$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseOutputTextDelta' from JSON`);
}
/** @internal */
export const Part2$inboundSchema = z.union([
    ResponseOutputText$inboundSchema,
    ReasoningTextContent$inboundSchema.and(z.object({ type: z.literal("reasoning_text") })),
    OpenAIResponsesRefusalContent$inboundSchema,
]);
export function part2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Part2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Part2' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseContentPartDone$inboundSchema = z
    .object({
    type: z.literal("response.content_part.done"),
    output_index: z.number(),
    item_id: z.string(),
    content_index: z.number(),
    part: z.union([
        ResponseOutputText$inboundSchema,
        ReasoningTextContent$inboundSchema.and(z.object({ type: z.literal("reasoning_text") })),
        OpenAIResponsesRefusalContent$inboundSchema,
    ]),
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "item_id": "itemId",
        "content_index": "contentIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseContentPartDoneFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseContentPartDone$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseContentPartDone' from JSON`);
}
/** @internal */
export const Part1$inboundSchema = z.union([
    ResponseOutputText$inboundSchema,
    ReasoningTextContent$inboundSchema.and(z.object({ type: z.literal("reasoning_text") })),
    OpenAIResponsesRefusalContent$inboundSchema,
]);
export function part1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Part1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Part1' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseContentPartAdded$inboundSchema = z
    .object({
    type: z.literal("response.content_part.added"),
    output_index: z.number(),
    item_id: z.string(),
    content_index: z.number(),
    part: z.union([
        ResponseOutputText$inboundSchema,
        ReasoningTextContent$inboundSchema.and(z.object({ type: z.literal("reasoning_text") })),
        OpenAIResponsesRefusalContent$inboundSchema,
    ]),
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "item_id": "itemId",
        "content_index": "contentIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseContentPartAddedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseContentPartAdded$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseContentPartAdded' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseOutputItemDone$inboundSchema = z.object({
    type: z.literal("response.output_item.done"),
    output_index: z.number(),
    item: ResponsesOutputItem$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseOutputItemDoneFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseOutputItemDone$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseOutputItemDone' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseOutputItemAdded$inboundSchema = z
    .object({
    type: z.literal("response.output_item.added"),
    output_index: z.number(),
    item: ResponsesOutputItem$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseOutputItemAddedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseOutputItemAdded$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseOutputItemAdded' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseFailed$inboundSchema = z.object({
    type: z.literal("response.failed"),
    response: OpenResponsesNonStreamingResponse$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseFailedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseFailed$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseFailed' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseIncomplete$inboundSchema = z.object({
    type: z.literal("response.incomplete"),
    response: OpenResponsesNonStreamingResponse$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseIncompleteFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseIncomplete$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseIncomplete' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseCompleted$inboundSchema = z.object({
    type: z.literal("response.completed"),
    response: OpenResponsesNonStreamingResponse$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseCompletedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseCompleted$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseCompleted' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseInProgress$inboundSchema = z.object({
    type: z.literal("response.in_progress"),
    response: OpenResponsesNonStreamingResponse$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseInProgressFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseInProgress$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseInProgress' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEventResponseCreated$inboundSchema = z.object({
    type: z.literal("response.created"),
    response: OpenResponsesNonStreamingResponse$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesStreamEventResponseCreatedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEventResponseCreated$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEventResponseCreated' from JSON`);
}
/** @internal */
export const OpenResponsesStreamEvent$inboundSchema = z.union([
    z.lazy(() => OpenResponsesStreamEventResponseCreated$inboundSchema),
    z.lazy(() => OpenResponsesStreamEventResponseInProgress$inboundSchema),
    z.lazy(() => OpenResponsesStreamEventResponseCompleted$inboundSchema),
    z.lazy(() => OpenResponsesStreamEventResponseIncomplete$inboundSchema),
    z.lazy(() => OpenResponsesStreamEventResponseFailed$inboundSchema),
    OpenResponsesErrorEvent$inboundSchema,
    z.lazy(() => OpenResponsesStreamEventResponseOutputItemAdded$inboundSchema),
    z.lazy(() => OpenResponsesStreamEventResponseOutputItemDone$inboundSchema),
    z.lazy(() => OpenResponsesStreamEventResponseContentPartAdded$inboundSchema),
    z.lazy(() => OpenResponsesStreamEventResponseContentPartDone$inboundSchema),
    z.lazy(() => OpenResponsesStreamEventResponseOutputTextDelta$inboundSchema),
    z.lazy(() => OpenResponsesStreamEventResponseOutputTextDone$inboundSchema),
    z.lazy(() => OpenResponsesStreamEventResponseRefusalDelta$inboundSchema),
    z.lazy(() => OpenResponsesStreamEventResponseRefusalDone$inboundSchema),
    z.lazy(() => OpenResponsesStreamEventResponseOutputTextAnnotationAdded$inboundSchema),
    z.lazy(() => OpenResponsesStreamEventResponseFunctionCallArgumentsDelta$inboundSchema),
    z.lazy(() => OpenResponsesStreamEventResponseFunctionCallArgumentsDone$inboundSchema),
    OpenResponsesReasoningDeltaEvent$inboundSchema,
    OpenResponsesReasoningDoneEvent$inboundSchema,
    OpenResponsesReasoningSummaryPartAddedEvent$inboundSchema,
    z.lazy(() => OpenResponsesStreamEventResponseReasoningSummaryPartDone$inboundSchema),
    OpenResponsesReasoningSummaryTextDeltaEvent$inboundSchema,
    OpenResponsesReasoningSummaryTextDoneEvent$inboundSchema,
    OpenResponsesImageGenCallInProgress$inboundSchema,
    OpenResponsesImageGenCallGenerating$inboundSchema,
    OpenResponsesImageGenCallPartialImage$inboundSchema,
    OpenResponsesImageGenCallCompleted$inboundSchema,
]);
export function openResponsesStreamEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesStreamEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesStreamEvent' from JSON`);
}
