// @openrouter/sdk/esm/models/openresponsesreasoningconfig.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: c2f9d074fbd4
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { OpenAIResponsesReasoningEffort$outboundSchema, } from "./openairesponsesreasoningeffort.js";
import { ReasoningSummaryVerbosity$outboundSchema, } from "./reasoningsummaryverbosity.js";
/** @internal */
export const OpenResponsesReasoningConfig$outboundSchema = z.object({
    effort: z.nullable(OpenAIResponsesReasoningEffort$outboundSchema).optional(),
    summary: ReasoningSummaryVerbosity$outboundSchema.optional(),
    maxTokens: z.nullable(z.number()).optional(),
    enabled: z.nullable(z.boolean()).optional(),
}).transform((v) => {
    return remap$(v, {
        maxTokens: "max_tokens",
    });
});
export function openResponsesReasoningConfigToJSON(openResponsesReasoningConfig) {
    return JSON.stringify(OpenResponsesReasoningConfig$outboundSchema.parse(openResponsesReasoningConfig));
}
