// @openrouter/sdk/esm/models/openresponsesnonstreamingresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ffd6f7198dd7
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { OpenAIResponsesIncompleteDetails$inboundSchema, } from "./openairesponsesincompletedetails.js";
import { OpenAIResponsesInputUnion$inboundSchema, } from "./openairesponsesinputunion.js";
import { OpenAIResponsesPrompt$inboundSchema, } from "./openairesponsesprompt.js";
import { OpenAIResponsesReasoningConfig$inboundSchema, } from "./openairesponsesreasoningconfig.js";
import { OpenAIResponsesResponseStatus$inboundSchema, } from "./openairesponsesresponsestatus.js";
import { OpenAIResponsesServiceTier$inboundSchema, } from "./openairesponsesservicetier.js";
import { OpenAIResponsesToolChoiceUnion$inboundSchema, } from "./openairesponsestoolchoiceunion.js";
import { OpenAIResponsesTruncation$inboundSchema, } from "./openairesponsestruncation.js";
import { OpenResponsesUsage$inboundSchema, } from "./openresponsesusage.js";
import { OpenResponsesWebSearch20250826Tool$inboundSchema, } from "./openresponseswebsearch20250826tool.js";
import { OpenResponsesWebSearchPreview20250311Tool$inboundSchema, } from "./openresponseswebsearchpreview20250311tool.js";
import { OpenResponsesWebSearchPreviewTool$inboundSchema, } from "./openresponseswebsearchpreviewtool.js";
import { OpenResponsesWebSearchTool$inboundSchema, } from "./openresponseswebsearchtool.js";
import { ResponsesErrorField$inboundSchema, } from "./responseserrorfield.js";
import { ResponsesOutputItem$inboundSchema, } from "./responsesoutputitem.js";
import { ResponseTextConfig$inboundSchema, } from "./responsetextconfig.js";
export const ObjectT = {
    Response: "response",
};
/** @internal */
export const ObjectT$inboundSchema = z.enum(ObjectT);
/** @internal */
export const OpenResponsesNonStreamingResponseToolFunction$inboundSchema = z.object({
    type: z.literal("function"),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    strict: z.nullable(z.boolean()).optional(),
    parameters: z.nullable(z.record(z.string(), z.nullable(z.any()))),
});
export function openResponsesNonStreamingResponseToolFunctionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesNonStreamingResponseToolFunction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesNonStreamingResponseToolFunction' from JSON`);
}
/** @internal */
export const OpenResponsesNonStreamingResponseToolUnion$inboundSchema = z.union([
    z.lazy(() => OpenResponsesNonStreamingResponseToolFunction$inboundSchema),
    OpenResponsesWebSearchPreviewTool$inboundSchema,
    OpenResponsesWebSearchPreview20250311Tool$inboundSchema,
    OpenResponsesWebSearchTool$inboundSchema,
    OpenResponsesWebSearch20250826Tool$inboundSchema,
]);
export function openResponsesNonStreamingResponseToolUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesNonStreamingResponseToolUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesNonStreamingResponseToolUnion' from JSON`);
}
/** @internal */
export const OpenResponsesNonStreamingResponse$inboundSchema = z.object({
    id: z.string(),
    object: ObjectT$inboundSchema,
    created_at: z.number(),
    model: z.string(),
    status: OpenAIResponsesResponseStatus$inboundSchema.optional(),
    output: z.array(ResponsesOutputItem$inboundSchema),
    user: z.nullable(z.string()).optional(),
    output_text: z.string().optional(),
    prompt_cache_key: z.nullable(z.string()).optional(),
    safety_identifier: z.nullable(z.string()).optional(),
    error: z.nullable(ResponsesErrorField$inboundSchema),
    incomplete_details: z.nullable(OpenAIResponsesIncompleteDetails$inboundSchema),
    usage: OpenResponsesUsage$inboundSchema.optional(),
    max_tool_calls: z.nullable(z.number()).optional(),
    top_logprobs: z.number().optional(),
    max_output_tokens: z.nullable(z.number()).optional(),
    temperature: z.nullable(z.number()),
    top_p: z.nullable(z.number()),
    instructions: z.nullable(OpenAIResponsesInputUnion$inboundSchema).optional(),
    metadata: z.nullable(z.record(z.string(), z.string())),
    tools: z.array(z.union([
        z.lazy(() => OpenResponsesNonStreamingResponseToolFunction$inboundSchema),
        OpenResponsesWebSearchPreviewTool$inboundSchema,
        OpenResponsesWebSearchPreview20250311Tool$inboundSchema,
        OpenResponsesWebSearchTool$inboundSchema,
        OpenResponsesWebSearch20250826Tool$inboundSchema,
    ])),
    tool_choice: OpenAIResponsesToolChoiceUnion$inboundSchema,
    parallel_tool_calls: z.boolean(),
    prompt: z.nullable(OpenAIResponsesPrompt$inboundSchema).optional(),
    background: z.nullable(z.boolean()).optional(),
    previous_response_id: z.nullable(z.string()).optional(),
    reasoning: z.nullable(OpenAIResponsesReasoningConfig$inboundSchema)
        .optional(),
    service_tier: z.nullable(OpenAIResponsesServiceTier$inboundSchema).optional(),
    store: z.boolean().optional(),
    truncation: z.nullable(OpenAIResponsesTruncation$inboundSchema).optional(),
    text: ResponseTextConfig$inboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "output_text": "outputText",
        "prompt_cache_key": "promptCacheKey",
        "safety_identifier": "safetyIdentifier",
        "incomplete_details": "incompleteDetails",
        "max_tool_calls": "maxToolCalls",
        "top_logprobs": "topLogprobs",
        "max_output_tokens": "maxOutputTokens",
        "top_p": "topP",
        "tool_choice": "toolChoice",
        "parallel_tool_calls": "parallelToolCalls",
        "previous_response_id": "previousResponseId",
        "service_tier": "serviceTier",
    });
});
export function openResponsesNonStreamingResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesNonStreamingResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesNonStreamingResponse' from JSON`);
}
