// @openrouter/sdk/esm/models/openresponsesreasoningsummarytextdoneevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0519f084cea8
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const OpenResponsesReasoningSummaryTextDoneEvent$inboundSchema = z.object({
    type: z.literal("response.reasoning_summary_text.done"),
    item_id: z.string(),
    output_index: z.number(),
    summary_index: z.number(),
    text: z.string(),
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "item_id": "itemId",
        "output_index": "outputIndex",
        "summary_index": "summaryIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesReasoningSummaryTextDoneEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesReasoningSummaryTextDoneEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesReasoningSummaryTextDoneEvent' from JSON`);
}
