// @openrouter/sdk/esm/models/openresponsesreasoningsummarypartaddedevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e83b9ca2ee64
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ReasoningSummaryText$inboundSchema, } from "./reasoningsummarytext.js";
/** @internal */
export const OpenResponsesReasoningSummaryPartAddedEvent$inboundSchema = z.object({
    type: z.literal("response.reasoning_summary_part.added"),
    output_index: z.number(),
    item_id: z.string(),
    summary_index: z.number(),
    part: ReasoningSummaryText$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "item_id": "itemId",
        "summary_index": "summaryIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesReasoningSummaryPartAddedEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesReasoningSummaryPartAddedEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesReasoningSummaryPartAddedEvent' from JSON`);
}
