// @openrouter/sdk/esm/models/openresponsesfunctioncalloutput.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5e801f7b2903
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OpenResponsesFunctionCallOutputType = {
    FunctionCallOutput: "function_call_output",
};
/** @internal */
export const OpenResponsesFunctionCallOutputType$outboundSchema = z.enum(OpenResponsesFunctionCallOutputType);
/** @internal */
export const OpenResponsesFunctionCallOutput$outboundSchema = z.object({
    type: OpenResponsesFunctionCallOutputType$outboundSchema,
    id: z.nullable(z.string()).optional(),
    callId: z.string(),
    output: z.string(),
    status: z.nullable(ToolCallStatus$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        callId: "call_id",
    });
});
export function openResponsesFunctionCallOutputToJSON(openResponsesFunctionCallOutput) {
    return JSON.stringify(OpenResponsesFunctionCallOutput$outboundSchema.parse(openResponsesFunctionCallOutput));
}
