// @openrouter/sdk/esm/models/openresponsesinputmessageitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: cccc4eb21a9b
 */
import * as z from "zod/v4";
import { ResponseInputAudio$outboundSchema, } from "./responseinputaudio.js";
import { ResponseInputFile$outboundSchema, } from "./responseinputfile.js";
import { ResponseInputImage$outboundSchema, } from "./responseinputimage.js";
import { ResponseInputText$outboundSchema, } from "./responseinputtext.js";
export const OpenResponsesInputMessageItemType = {
    Message: "message",
};
export const OpenResponsesInputMessageItemRoleDeveloper = {
    Developer: "developer",
};
export const OpenResponsesInputMessageItemRoleSystem = {
    System: "system",
};
export const OpenResponsesInputMessageItemRoleUser = {
    User: "user",
};
/** @internal */
export const OpenResponsesInputMessageItemType$outboundSchema = z.enum(OpenResponsesInputMessageItemType);
/** @internal */
export const OpenResponsesInputMessageItemRoleDeveloper$outboundSchema = z.enum(OpenResponsesInputMessageItemRoleDeveloper);
/** @internal */
export const OpenResponsesInputMessageItemRoleSystem$outboundSchema = z.enum(OpenResponsesInputMessageItemRoleSystem);
/** @internal */
export const OpenResponsesInputMessageItemRoleUser$outboundSchema = z.enum(OpenResponsesInputMessageItemRoleUser);
/** @internal */
export const OpenResponsesInputMessageItemRoleUnion$outboundSchema = z.union([
    OpenResponsesInputMessageItemRoleUser$outboundSchema,
    OpenResponsesInputMessageItemRoleSystem$outboundSchema,
    OpenResponsesInputMessageItemRoleDeveloper$outboundSchema,
]);
export function openResponsesInputMessageItemRoleUnionToJSON(openResponsesInputMessageItemRoleUnion) {
    return JSON.stringify(OpenResponsesInputMessageItemRoleUnion$outboundSchema.parse(openResponsesInputMessageItemRoleUnion));
}
/** @internal */
export const OpenResponsesInputMessageItemContent$outboundSchema = z.union([
    ResponseInputText$outboundSchema,
    ResponseInputImage$outboundSchema,
    ResponseInputFile$outboundSchema,
    ResponseInputAudio$outboundSchema,
]);
export function openResponsesInputMessageItemContentToJSON(openResponsesInputMessageItemContent) {
    return JSON.stringify(OpenResponsesInputMessageItemContent$outboundSchema.parse(openResponsesInputMessageItemContent));
}
/** @internal */
export const OpenResponsesInputMessageItem$outboundSchema = z.object({
    id: z.string().optional(),
    type: OpenResponsesInputMessageItemType$outboundSchema.optional(),
    role: z.union([
        OpenResponsesInputMessageItemRoleUser$outboundSchema,
        OpenResponsesInputMessageItemRoleSystem$outboundSchema,
        OpenResponsesInputMessageItemRoleDeveloper$outboundSchema,
    ]),
    content: z.array(z.union([
        ResponseInputText$outboundSchema,
        ResponseInputImage$outboundSchema,
        ResponseInputFile$outboundSchema,
        ResponseInputAudio$outboundSchema,
    ])),
});
export function openResponsesInputMessageItemToJSON(openResponsesInputMessageItem) {
    return JSON.stringify(OpenResponsesInputMessageItem$outboundSchema.parse(openResponsesInputMessageItem));
}
