// @openrouter/sdk/esm/models/openresponseswebsearchpreviewtool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7d594dd37b72
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ResponsesSearchContextSize$inboundSchema, ResponsesSearchContextSize$outboundSchema, } from "./responsessearchcontextsize.js";
import { WebSearchPreviewToolUserLocation$inboundSchema, WebSearchPreviewToolUserLocation$outboundSchema, } from "./websearchpreviewtooluserlocation.js";
/** @internal */
export const OpenResponsesWebSearchPreviewTool$inboundSchema = z.object({
    type: z.literal("web_search_preview"),
    search_context_size: ResponsesSearchContextSize$inboundSchema.optional(),
    user_location: z.nullable(WebSearchPreviewToolUserLocation$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "search_context_size": "searchContextSize",
        "user_location": "userLocation",
    });
});
/** @internal */
export const OpenResponsesWebSearchPreviewTool$outboundSchema = z.object({
    type: z.literal("web_search_preview"),
    searchContextSize: ResponsesSearchContextSize$outboundSchema.optional(),
    userLocation: z.nullable(WebSearchPreviewToolUserLocation$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        searchContextSize: "search_context_size",
        userLocation: "user_location",
    });
});
export function openResponsesWebSearchPreviewToolToJSON(openResponsesWebSearchPreviewTool) {
    return JSON.stringify(OpenResponsesWebSearchPreviewTool$outboundSchema.parse(openResponsesWebSearchPreviewTool));
}
export function openResponsesWebSearchPreviewToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesWebSearchPreviewTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesWebSearchPreviewTool' from JSON`);
}
