// @openrouter/sdk/esm/models/openresponseswebsearchpreview20250311tool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: d450efb8b660
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ResponsesSearchContextSize$inboundSchema, ResponsesSearchContextSize$outboundSchema, } from "./responsessearchcontextsize.js";
import { WebSearchPreviewToolUserLocation$inboundSchema, WebSearchPreviewToolUserLocation$outboundSchema, } from "./websearchpreviewtooluserlocation.js";
/** @internal */
export const OpenResponsesWebSearchPreview20250311Tool$inboundSchema = z.object({
    type: z.literal("web_search_preview_2025_03_11"),
    search_context_size: ResponsesSearchContextSize$inboundSchema.optional(),
    user_location: z.nullable(WebSearchPreviewToolUserLocation$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "search_context_size": "searchContextSize",
        "user_location": "userLocation",
    });
});
/** @internal */
export const OpenResponsesWebSearchPreview20250311Tool$outboundSchema = z.object({
    type: z.literal("web_search_preview_2025_03_11"),
    searchContextSize: ResponsesSearchContextSize$outboundSchema.optional(),
    userLocation: z.nullable(WebSearchPreviewToolUserLocation$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        searchContextSize: "search_context_size",
        userLocation: "user_location",
    });
});
export function openResponsesWebSearchPreview20250311ToolToJSON(openResponsesWebSearchPreview20250311Tool) {
    return JSON.stringify(OpenResponsesWebSearchPreview20250311Tool$outboundSchema.parse(openResponsesWebSearchPreview20250311Tool));
}
export function openResponsesWebSearchPreview20250311ToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesWebSearchPreview20250311Tool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesWebSearchPreview20250311Tool' from JSON`);
}
