// @openrouter/sdk/esm/models/openresponsesreasoningsummarytextdeltaevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 61ee69f0a41f
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const OpenResponsesReasoningSummaryTextDeltaEvent$inboundSchema = z.object({
    type: z.literal("response.reasoning_summary_text.delta"),
    item_id: z.string(),
    output_index: z.number(),
    summary_index: z.number(),
    delta: z.string(),
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "item_id": "itemId",
        "output_index": "outputIndex",
        "summary_index": "summaryIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function openResponsesReasoningSummaryTextDeltaEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesReasoningSummaryTextDeltaEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesReasoningSummaryTextDeltaEvent' from JSON`);
}
