// @openrouter/sdk/esm/models/openairesponsesinputunion.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0b061e126936
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { OutputItemImageGenerationCall$inboundSchema, } from "./outputitemimagegenerationcall.js";
import { OutputMessage$inboundSchema } from "./outputmessage.js";
import { ResponseInputAudio$inboundSchema, } from "./responseinputaudio.js";
import { ResponseInputFile$inboundSchema, } from "./responseinputfile.js";
import { ResponseInputImage$inboundSchema, } from "./responseinputimage.js";
import { ResponseInputText$inboundSchema, } from "./responseinputtext.js";
import { ToolCallStatus$inboundSchema, } from "./toolcallstatus.js";
export const OpenAIResponsesInputTypeFunctionCall = {
    FunctionCall: "function_call",
};
export const OpenAIResponsesInputTypeFunctionCallOutput = {
    FunctionCallOutput: "function_call_output",
};
export const OpenAIResponsesInputTypeMessage2 = {
    Message: "message",
};
export const OpenAIResponsesInputRoleDeveloper2 = {
    Developer: "developer",
};
export const OpenAIResponsesInputRoleSystem2 = {
    System: "system",
};
export const OpenAIResponsesInputRoleUser2 = {
    User: "user",
};
export const OpenAIResponsesInputTypeMessage1 = {
    Message: "message",
};
export const OpenAIResponsesInputRoleDeveloper1 = {
    Developer: "developer",
};
export const OpenAIResponsesInputRoleAssistant = {
    Assistant: "assistant",
};
export const OpenAIResponsesInputRoleSystem1 = {
    System: "system",
};
export const OpenAIResponsesInputRoleUser1 = {
    User: "user",
};
/** @internal */
export const OpenAIResponsesInputTypeFunctionCall$inboundSchema = z.enum(OpenAIResponsesInputTypeFunctionCall);
/** @internal */
export const OpenAIResponsesInputFunctionCall$inboundSchema = z.object({
    type: OpenAIResponsesInputTypeFunctionCall$inboundSchema,
    call_id: z.string(),
    name: z.string(),
    arguments: z.string(),
    id: z.string().optional(),
    status: z.nullable(ToolCallStatus$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
    });
});
export function openAIResponsesInputFunctionCallFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponsesInputFunctionCall$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponsesInputFunctionCall' from JSON`);
}
/** @internal */
export const OpenAIResponsesInputTypeFunctionCallOutput$inboundSchema = z.enum(OpenAIResponsesInputTypeFunctionCallOutput);
/** @internal */
export const OpenAIResponsesInputFunctionCallOutput$inboundSchema = z.object({
    type: OpenAIResponsesInputTypeFunctionCallOutput$inboundSchema,
    id: z.nullable(z.string()).optional(),
    call_id: z.string(),
    output: z.string(),
    status: z.nullable(ToolCallStatus$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
    });
});
export function openAIResponsesInputFunctionCallOutputFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponsesInputFunctionCallOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponsesInputFunctionCallOutput' from JSON`);
}
/** @internal */
export const OpenAIResponsesInputTypeMessage2$inboundSchema = z.enum(OpenAIResponsesInputTypeMessage2);
/** @internal */
export const OpenAIResponsesInputRoleDeveloper2$inboundSchema = z.enum(OpenAIResponsesInputRoleDeveloper2);
/** @internal */
export const OpenAIResponsesInputRoleSystem2$inboundSchema = z.enum(OpenAIResponsesInputRoleSystem2);
/** @internal */
export const OpenAIResponsesInputRoleUser2$inboundSchema = z.enum(OpenAIResponsesInputRoleUser2);
/** @internal */
export const OpenAIResponsesInputRoleUnion2$inboundSchema = z.union([
    OpenAIResponsesInputRoleUser2$inboundSchema,
    OpenAIResponsesInputRoleSystem2$inboundSchema,
    OpenAIResponsesInputRoleDeveloper2$inboundSchema,
]);
export function openAIResponsesInputRoleUnion2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponsesInputRoleUnion2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponsesInputRoleUnion2' from JSON`);
}
/** @internal */
export const OpenAIResponsesInputContent3$inboundSchema = z.union([
    ResponseInputText$inboundSchema,
    ResponseInputImage$inboundSchema,
    ResponseInputFile$inboundSchema,
    ResponseInputAudio$inboundSchema,
]);
export function openAIResponsesInputContent3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponsesInputContent3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponsesInputContent3' from JSON`);
}
/** @internal */
export const OpenAIResponsesInputMessage2$inboundSchema = z.object({
    id: z.string(),
    type: OpenAIResponsesInputTypeMessage2$inboundSchema.optional(),
    role: z.union([
        OpenAIResponsesInputRoleUser2$inboundSchema,
        OpenAIResponsesInputRoleSystem2$inboundSchema,
        OpenAIResponsesInputRoleDeveloper2$inboundSchema,
    ]),
    content: z.array(z.union([
        ResponseInputText$inboundSchema,
        ResponseInputImage$inboundSchema,
        ResponseInputFile$inboundSchema,
        ResponseInputAudio$inboundSchema,
    ])),
});
export function openAIResponsesInputMessage2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponsesInputMessage2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponsesInputMessage2' from JSON`);
}
/** @internal */
export const OpenAIResponsesInputTypeMessage1$inboundSchema = z.enum(OpenAIResponsesInputTypeMessage1);
/** @internal */
export const OpenAIResponsesInputRoleDeveloper1$inboundSchema = z.enum(OpenAIResponsesInputRoleDeveloper1);
/** @internal */
export const OpenAIResponsesInputRoleAssistant$inboundSchema = z.enum(OpenAIResponsesInputRoleAssistant);
/** @internal */
export const OpenAIResponsesInputRoleSystem1$inboundSchema = z.enum(OpenAIResponsesInputRoleSystem1);
/** @internal */
export const OpenAIResponsesInputRoleUser1$inboundSchema = z.enum(OpenAIResponsesInputRoleUser1);
/** @internal */
export const OpenAIResponsesInputRoleUnion1$inboundSchema = z.union([
    OpenAIResponsesInputRoleUser1$inboundSchema,
    OpenAIResponsesInputRoleSystem1$inboundSchema,
    OpenAIResponsesInputRoleAssistant$inboundSchema,
    OpenAIResponsesInputRoleDeveloper1$inboundSchema,
]);
export function openAIResponsesInputRoleUnion1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponsesInputRoleUnion1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponsesInputRoleUnion1' from JSON`);
}
/** @internal */
export const OpenAIResponsesInputContent1$inboundSchema = z.union([
    ResponseInputText$inboundSchema,
    ResponseInputImage$inboundSchema,
    ResponseInputFile$inboundSchema,
    ResponseInputAudio$inboundSchema,
]);
export function openAIResponsesInputContent1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponsesInputContent1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponsesInputContent1' from JSON`);
}
/** @internal */
export const OpenAIResponsesInputContent2$inboundSchema = z.union([
    z.array(z.union([
        ResponseInputText$inboundSchema,
        ResponseInputImage$inboundSchema,
        ResponseInputFile$inboundSchema,
        ResponseInputAudio$inboundSchema,
    ])),
    z.string(),
]);
export function openAIResponsesInputContent2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponsesInputContent2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponsesInputContent2' from JSON`);
}
/** @internal */
export const OpenAIResponsesInputMessage1$inboundSchema = z.object({
    type: OpenAIResponsesInputTypeMessage1$inboundSchema.optional(),
    role: z.union([
        OpenAIResponsesInputRoleUser1$inboundSchema,
        OpenAIResponsesInputRoleSystem1$inboundSchema,
        OpenAIResponsesInputRoleAssistant$inboundSchema,
        OpenAIResponsesInputRoleDeveloper1$inboundSchema,
    ]),
    content: z.union([
        z.array(z.union([
            ResponseInputText$inboundSchema,
            ResponseInputImage$inboundSchema,
            ResponseInputFile$inboundSchema,
            ResponseInputAudio$inboundSchema,
        ])),
        z.string(),
    ]),
});
export function openAIResponsesInputMessage1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponsesInputMessage1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponsesInputMessage1' from JSON`);
}
/** @internal */
export const OpenAIResponsesInputUnion1$inboundSchema = z.union([
    z.lazy(() => OpenAIResponsesInputFunctionCall$inboundSchema),
    OutputMessage$inboundSchema,
    z.lazy(() => OpenAIResponsesInputMessage2$inboundSchema),
    z.lazy(() => OpenAIResponsesInputFunctionCallOutput$inboundSchema),
    OutputItemImageGenerationCall$inboundSchema,
    z.lazy(() => OpenAIResponsesInputMessage1$inboundSchema),
]);
export function openAIResponsesInputUnion1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponsesInputUnion1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponsesInputUnion1' from JSON`);
}
/** @internal */
export const OpenAIResponsesInputUnion$inboundSchema = z.union([
    z.string(),
    z.array(z.union([
        z.lazy(() => OpenAIResponsesInputFunctionCall$inboundSchema),
        OutputMessage$inboundSchema,
        z.lazy(() => OpenAIResponsesInputMessage2$inboundSchema),
        z.lazy(() => OpenAIResponsesInputFunctionCallOutput$inboundSchema),
        OutputItemImageGenerationCall$inboundSchema,
        z.lazy(() => OpenAIResponsesInputMessage1$inboundSchema),
    ])),
    z.any(),
]);
export function openAIResponsesInputUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponsesInputUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponsesInputUnion' from JSON`);
}
