// @openrouter/sdk/esm/models/openresponseswebsearchtool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 510ab58460c2
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ResponsesSearchContextSize$inboundSchema, ResponsesSearchContextSize$outboundSchema, } from "./responsessearchcontextsize.js";
import { ResponsesWebSearchUserLocation$inboundSchema, ResponsesWebSearchUserLocation$outboundSchema, } from "./responseswebsearchuserlocation.js";
/** @internal */
export const OpenResponsesWebSearchToolFilters$inboundSchema = z.object({
    allowed_domains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        "allowed_domains": "allowedDomains",
    });
});
/** @internal */
export const OpenResponsesWebSearchToolFilters$outboundSchema = z.object({
    allowedDomains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        allowedDomains: "allowed_domains",
    });
});
export function openResponsesWebSearchToolFiltersToJSON(openResponsesWebSearchToolFilters) {
    return JSON.stringify(OpenResponsesWebSearchToolFilters$outboundSchema.parse(openResponsesWebSearchToolFilters));
}
export function openResponsesWebSearchToolFiltersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesWebSearchToolFilters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesWebSearchToolFilters' from JSON`);
}
/** @internal */
export const OpenResponsesWebSearchTool$inboundSchema = z.object({
    type: z.literal("web_search"),
    filters: z.nullable(z.lazy(() => OpenResponsesWebSearchToolFilters$inboundSchema)).optional(),
    search_context_size: ResponsesSearchContextSize$inboundSchema.optional(),
    user_location: z.nullable(ResponsesWebSearchUserLocation$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "search_context_size": "searchContextSize",
        "user_location": "userLocation",
    });
});
/** @internal */
export const OpenResponsesWebSearchTool$outboundSchema = z.object({
    type: z.literal("web_search"),
    filters: z.nullable(z.lazy(() => OpenResponsesWebSearchToolFilters$outboundSchema)).optional(),
    searchContextSize: ResponsesSearchContextSize$outboundSchema.optional(),
    userLocation: z.nullable(ResponsesWebSearchUserLocation$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        searchContextSize: "search_context_size",
        userLocation: "user_location",
    });
});
export function openResponsesWebSearchToolToJSON(openResponsesWebSearchTool) {
    return JSON.stringify(OpenResponsesWebSearchTool$outboundSchema.parse(openResponsesWebSearchTool));
}
export function openResponsesWebSearchToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesWebSearchTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesWebSearchTool' from JSON`);
}
