// @openrouter/sdk/esm/models/openresponseswebsearch20250826tool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7650f38c7184
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ResponsesSearchContextSize$inboundSchema, ResponsesSearchContextSize$outboundSchema, } from "./responsessearchcontextsize.js";
import { ResponsesWebSearchUserLocation$inboundSchema, ResponsesWebSearchUserLocation$outboundSchema, } from "./responseswebsearchuserlocation.js";
/** @internal */
export const OpenResponsesWebSearch20250826ToolFilters$inboundSchema = z.object({
    allowed_domains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        "allowed_domains": "allowedDomains",
    });
});
/** @internal */
export const OpenResponsesWebSearch20250826ToolFilters$outboundSchema = z.object({
    allowedDomains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        allowedDomains: "allowed_domains",
    });
});
export function openResponsesWebSearch20250826ToolFiltersToJSON(openResponsesWebSearch20250826ToolFilters) {
    return JSON.stringify(OpenResponsesWebSearch20250826ToolFilters$outboundSchema.parse(openResponsesWebSearch20250826ToolFilters));
}
export function openResponsesWebSearch20250826ToolFiltersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesWebSearch20250826ToolFilters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesWebSearch20250826ToolFilters' from JSON`);
}
/** @internal */
export const OpenResponsesWebSearch20250826Tool$inboundSchema = z.object({
    type: z.literal("web_search_2025_08_26"),
    filters: z.nullable(z.lazy(() => OpenResponsesWebSearch20250826ToolFilters$inboundSchema)).optional(),
    search_context_size: ResponsesSearchContextSize$inboundSchema.optional(),
    user_location: z.nullable(ResponsesWebSearchUserLocation$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "search_context_size": "searchContextSize",
        "user_location": "userLocation",
    });
});
/** @internal */
export const OpenResponsesWebSearch20250826Tool$outboundSchema = z.object({
    type: z.literal("web_search_2025_08_26"),
    filters: z.nullable(z.lazy(() => OpenResponsesWebSearch20250826ToolFilters$outboundSchema)).optional(),
    searchContextSize: ResponsesSearchContextSize$outboundSchema.optional(),
    userLocation: z.nullable(ResponsesWebSearchUserLocation$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        searchContextSize: "search_context_size",
        userLocation: "user_location",
    });
});
export function openResponsesWebSearch20250826ToolToJSON(openResponsesWebSearch20250826Tool) {
    return JSON.stringify(OpenResponsesWebSearch20250826Tool$outboundSchema.parse(openResponsesWebSearch20250826Tool));
}
export function openResponsesWebSearch20250826ToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesWebSearch20250826Tool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesWebSearch20250826Tool' from JSON`);
}
