// @openrouter/sdk/esm/models/openairesponsesprompt.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: c583d897e9d2
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ResponseInputFile$inboundSchema, ResponseInputFile$outboundSchema, } from "./responseinputfile.js";
import { ResponseInputImage$inboundSchema, ResponseInputImage$outboundSchema, } from "./responseinputimage.js";
import { ResponseInputText$inboundSchema, ResponseInputText$outboundSchema, } from "./responseinputtext.js";
/** @internal */
export const Variables$inboundSchema = z.union([
    ResponseInputText$inboundSchema,
    ResponseInputImage$inboundSchema,
    ResponseInputFile$inboundSchema,
    z.string(),
]);
/** @internal */
export const Variables$outboundSchema = z.union([
    ResponseInputText$outboundSchema,
    ResponseInputImage$outboundSchema,
    ResponseInputFile$outboundSchema,
    z.string(),
]);
export function variablesToJSON(variables) {
    return JSON.stringify(Variables$outboundSchema.parse(variables));
}
export function variablesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Variables$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Variables' from JSON`);
}
/** @internal */
export const OpenAIResponsesPrompt$inboundSchema = z.object({
    id: z.string(),
    variables: z.nullable(z.record(z.string(), z.union([
        ResponseInputText$inboundSchema,
        ResponseInputImage$inboundSchema,
        ResponseInputFile$inboundSchema,
        z.string(),
    ]))).optional(),
});
/** @internal */
export const OpenAIResponsesPrompt$outboundSchema = z.object({
    id: z.string(),
    variables: z.nullable(z.record(z.string(), z.union([
        ResponseInputText$outboundSchema,
        ResponseInputImage$outboundSchema,
        ResponseInputFile$outboundSchema,
        z.string(),
    ]))).optional(),
});
export function openAIResponsesPromptToJSON(openAIResponsesPrompt) {
    return JSON.stringify(OpenAIResponsesPrompt$outboundSchema.parse(openAIResponsesPrompt));
}
export function openAIResponsesPromptFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIResponsesPrompt$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIResponsesPrompt' from JSON`);
}
