// @openrouter/sdk/esm/models/outputmessage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: fa128cf87d51
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { OpenAIResponsesRefusalContent$inboundSchema, } from "./openairesponsesrefusalcontent.js";
import { ResponseOutputText$inboundSchema, } from "./responseoutputtext.js";
export const OutputMessageRole = {
    Assistant: "assistant",
};
export const OutputMessageType = {
    Message: "message",
};
export const OutputMessageStatusInProgress = {
    InProgress: "in_progress",
};
export const OutputMessageStatusIncomplete = {
    Incomplete: "incomplete",
};
export const OutputMessageStatusCompleted = {
    Completed: "completed",
};
/** @internal */
export const OutputMessageRole$inboundSchema = z.enum(OutputMessageRole);
/** @internal */
export const OutputMessageType$inboundSchema = z.enum(OutputMessageType);
/** @internal */
export const OutputMessageStatusInProgress$inboundSchema = z.enum(OutputMessageStatusInProgress);
/** @internal */
export const OutputMessageStatusIncomplete$inboundSchema = z.enum(OutputMessageStatusIncomplete);
/** @internal */
export const OutputMessageStatusCompleted$inboundSchema = z.enum(OutputMessageStatusCompleted);
/** @internal */
export const OutputMessageStatusUnion$inboundSchema = z.union([
    OutputMessageStatusCompleted$inboundSchema,
    OutputMessageStatusIncomplete$inboundSchema,
    OutputMessageStatusInProgress$inboundSchema,
]);
export function outputMessageStatusUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputMessageStatusUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputMessageStatusUnion' from JSON`);
}
/** @internal */
export const OutputMessageContent$inboundSchema = z.union([
    ResponseOutputText$inboundSchema,
    OpenAIResponsesRefusalContent$inboundSchema,
]);
export function outputMessageContentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputMessageContent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputMessageContent' from JSON`);
}
/** @internal */
export const OutputMessage$inboundSchema = z
    .object({
    id: z.string(),
    role: OutputMessageRole$inboundSchema,
    type: OutputMessageType$inboundSchema,
    status: z.union([
        OutputMessageStatusCompleted$inboundSchema,
        OutputMessageStatusIncomplete$inboundSchema,
        OutputMessageStatusInProgress$inboundSchema,
    ]).optional(),
    content: z.array(z.union([
        ResponseOutputText$inboundSchema,
        OpenAIResponsesRefusalContent$inboundSchema,
    ])),
});
export function outputMessageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputMessage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputMessage' from JSON`);
}
