// @openrouter/sdk/esm/models/reasoningsummarytext.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 9c06c18de6d6
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
export const ReasoningSummaryTextType = {
    SummaryText: "summary_text",
};
/** @internal */
export const ReasoningSummaryTextType$inboundSchema = z.enum(ReasoningSummaryTextType);
/** @internal */
export const ReasoningSummaryTextType$outboundSchema = ReasoningSummaryTextType$inboundSchema;
/** @internal */
export const ReasoningSummaryText$inboundSchema = z.object({
    type: ReasoningSummaryTextType$inboundSchema,
    text: z.string(),
});
/** @internal */
export const ReasoningSummaryText$outboundSchema = z.object({
    type: ReasoningSummaryTextType$outboundSchema,
    text: z.string(),
});
export function reasoningSummaryTextToJSON(reasoningSummaryText) {
    return JSON.stringify(ReasoningSummaryText$outboundSchema.parse(reasoningSummaryText));
}
export function reasoningSummaryTextFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ReasoningSummaryText$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ReasoningSummaryText' from JSON`);
}
