// @openrouter/sdk/esm/models/responseinputimage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 9c7a50d121df
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const ResponseInputImageDetail = {
    Auto: "auto",
    High: "high",
    Low: "low",
};
/** @internal */
export const ResponseInputImageDetail$inboundSchema = openEnums.inboundSchema(ResponseInputImageDetail);
/** @internal */
export const ResponseInputImageDetail$outboundSchema = openEnums.outboundSchema(ResponseInputImageDetail);
/** @internal */
export const ResponseInputImage$inboundSchema = z.object({
    type: z.literal("input_image"),
    detail: ResponseInputImageDetail$inboundSchema,
    image_url: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "image_url": "imageUrl",
    });
});
/** @internal */
export const ResponseInputImage$outboundSchema = z.object({
    type: z.literal("input_image"),
    detail: ResponseInputImageDetail$outboundSchema,
    imageUrl: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        imageUrl: "image_url",
    });
});
export function responseInputImageToJSON(responseInputImage) {
    return JSON.stringify(ResponseInputImage$outboundSchema.parse(responseInputImage));
}
export function responseInputImageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponseInputImage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponseInputImage' from JSON`);
}
