// @openrouter/sdk/esm/models/responseswebsearchcalloutput.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: c1c0ab68e92c
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { WebSearchStatus$inboundSchema, WebSearchStatus$outboundSchema, } from "./websearchstatus.js";
export const ResponsesWebSearchCallOutputType = {
    WebSearchCall: "web_search_call",
};
/** @internal */
export const ResponsesWebSearchCallOutputType$inboundSchema = z.enum(ResponsesWebSearchCallOutputType);
/** @internal */
export const ResponsesWebSearchCallOutputType$outboundSchema = ResponsesWebSearchCallOutputType$inboundSchema;
/** @internal */
export const ResponsesWebSearchCallOutput$inboundSchema = z.object({
    type: ResponsesWebSearchCallOutputType$inboundSchema,
    id: z.string(),
    status: WebSearchStatus$inboundSchema,
});
/** @internal */
export const ResponsesWebSearchCallOutput$outboundSchema = z.object({
    type: ResponsesWebSearchCallOutputType$outboundSchema,
    id: z.string(),
    status: WebSearchStatus$outboundSchema,
});
export function responsesWebSearchCallOutputToJSON(responsesWebSearchCallOutput) {
    return JSON.stringify(ResponsesWebSearchCallOutput$outboundSchema.parse(responsesWebSearchCallOutput));
}
export function responsesWebSearchCallOutputFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesWebSearchCallOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesWebSearchCallOutput' from JSON`);
}
