// @openrouter/sdk/esm/models/responsesformattextjsonschemaconfig.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 631c61a1b658
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const ResponsesFormatTextJSONSchemaConfig$inboundSchema = z.object({
    type: z.literal("json_schema"),
    name: z.string(),
    description: z.string().optional(),
    strict: z.nullable(z.boolean()).optional(),
    schema: z.record(z.string(), z.nullable(z.any())),
});
/** @internal */
export const ResponsesFormatTextJSONSchemaConfig$outboundSchema = z.object({
    type: z.literal("json_schema"),
    name: z.string(),
    description: z.string().optional(),
    strict: z.nullable(z.boolean()).optional(),
    schema: z.record(z.string(), z.nullable(z.any())),
});
export function responsesFormatTextJSONSchemaConfigToJSON(responsesFormatTextJSONSchemaConfig) {
    return JSON.stringify(ResponsesFormatTextJSONSchemaConfig$outboundSchema.parse(responsesFormatTextJSONSchemaConfig));
}
export function responsesFormatTextJSONSchemaConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesFormatTextJSONSchemaConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesFormatTextJSONSchemaConfig' from JSON`);
}
