// @openrouter/sdk/esm/models/responsesoutputitemfunctioncall.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ff553c6d83b4
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const ResponsesOutputItemFunctionCallType = {
    FunctionCall: "function_call",
};
export const ResponsesOutputItemFunctionCallStatusInProgress = {
    InProgress: "in_progress",
};
export const ResponsesOutputItemFunctionCallStatusIncomplete = {
    Incomplete: "incomplete",
};
export const ResponsesOutputItemFunctionCallStatusCompleted = {
    Completed: "completed",
};
/** @internal */
export const ResponsesOutputItemFunctionCallType$inboundSchema = z.enum(ResponsesOutputItemFunctionCallType);
/** @internal */
export const ResponsesOutputItemFunctionCallType$outboundSchema = ResponsesOutputItemFunctionCallType$inboundSchema;
/** @internal */
export const ResponsesOutputItemFunctionCallStatusInProgress$inboundSchema = z.enum(ResponsesOutputItemFunctionCallStatusInProgress);
/** @internal */
export const ResponsesOutputItemFunctionCallStatusInProgress$outboundSchema = ResponsesOutputItemFunctionCallStatusInProgress$inboundSchema;
/** @internal */
export const ResponsesOutputItemFunctionCallStatusIncomplete$inboundSchema = z.enum(ResponsesOutputItemFunctionCallStatusIncomplete);
/** @internal */
export const ResponsesOutputItemFunctionCallStatusIncomplete$outboundSchema = ResponsesOutputItemFunctionCallStatusIncomplete$inboundSchema;
/** @internal */
export const ResponsesOutputItemFunctionCallStatusCompleted$inboundSchema = z.enum(ResponsesOutputItemFunctionCallStatusCompleted);
/** @internal */
export const ResponsesOutputItemFunctionCallStatusCompleted$outboundSchema = ResponsesOutputItemFunctionCallStatusCompleted$inboundSchema;
/** @internal */
export const ResponsesOutputItemFunctionCallStatusUnion$inboundSchema = z.union([
    ResponsesOutputItemFunctionCallStatusCompleted$inboundSchema,
    ResponsesOutputItemFunctionCallStatusIncomplete$inboundSchema,
    ResponsesOutputItemFunctionCallStatusInProgress$inboundSchema,
]);
/** @internal */
export const ResponsesOutputItemFunctionCallStatusUnion$outboundSchema = z.union([
    ResponsesOutputItemFunctionCallStatusCompleted$outboundSchema,
    ResponsesOutputItemFunctionCallStatusIncomplete$outboundSchema,
    ResponsesOutputItemFunctionCallStatusInProgress$outboundSchema,
]);
export function responsesOutputItemFunctionCallStatusUnionToJSON(responsesOutputItemFunctionCallStatusUnion) {
    return JSON.stringify(ResponsesOutputItemFunctionCallStatusUnion$outboundSchema.parse(responsesOutputItemFunctionCallStatusUnion));
}
export function responsesOutputItemFunctionCallStatusUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesOutputItemFunctionCallStatusUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesOutputItemFunctionCallStatusUnion' from JSON`);
}
/** @internal */
export const ResponsesOutputItemFunctionCall$inboundSchema = z.object({
    type: ResponsesOutputItemFunctionCallType$inboundSchema,
    id: z.string().optional(),
    name: z.string(),
    arguments: z.string(),
    call_id: z.string(),
    status: z.union([
        ResponsesOutputItemFunctionCallStatusCompleted$inboundSchema,
        ResponsesOutputItemFunctionCallStatusIncomplete$inboundSchema,
        ResponsesOutputItemFunctionCallStatusInProgress$inboundSchema,
    ]).optional(),
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
    });
});
/** @internal */
export const ResponsesOutputItemFunctionCall$outboundSchema = z.object({
    type: ResponsesOutputItemFunctionCallType$outboundSchema,
    id: z.string().optional(),
    name: z.string(),
    arguments: z.string(),
    callId: z.string(),
    status: z.union([
        ResponsesOutputItemFunctionCallStatusCompleted$outboundSchema,
        ResponsesOutputItemFunctionCallStatusIncomplete$outboundSchema,
        ResponsesOutputItemFunctionCallStatusInProgress$outboundSchema,
    ]).optional(),
}).transform((v) => {
    return remap$(v, {
        callId: "call_id",
    });
});
export function responsesOutputItemFunctionCallToJSON(responsesOutputItemFunctionCall) {
    return JSON.stringify(ResponsesOutputItemFunctionCall$outboundSchema.parse(responsesOutputItemFunctionCall));
}
export function responsesOutputItemFunctionCallFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesOutputItemFunctionCall$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesOutputItemFunctionCall' from JSON`);
}
