// @openrouter/sdk/esm/models/responsesimagegenerationcall.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 51f0104f0dbf
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ImageGenerationStatus$inboundSchema, ImageGenerationStatus$outboundSchema, } from "./imagegenerationstatus.js";
export const ResponsesImageGenerationCallType = {
    ImageGenerationCall: "image_generation_call",
};
/** @internal */
export const ResponsesImageGenerationCallType$inboundSchema = z.enum(ResponsesImageGenerationCallType);
/** @internal */
export const ResponsesImageGenerationCallType$outboundSchema = ResponsesImageGenerationCallType$inboundSchema;
/** @internal */
export const ResponsesImageGenerationCall$inboundSchema = z.object({
    type: ResponsesImageGenerationCallType$inboundSchema,
    id: z.string(),
    result: z.nullable(z.string()).default(null),
    status: ImageGenerationStatus$inboundSchema,
});
/** @internal */
export const ResponsesImageGenerationCall$outboundSchema = z.object({
    type: ResponsesImageGenerationCallType$outboundSchema,
    id: z.string(),
    result: z.nullable(z.string()).default(null),
    status: ImageGenerationStatus$outboundSchema,
});
export function responsesImageGenerationCallToJSON(responsesImageGenerationCall) {
    return JSON.stringify(ResponsesImageGenerationCall$outboundSchema.parse(responsesImageGenerationCall));
}
export function responsesImageGenerationCallFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesImageGenerationCall$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesImageGenerationCall' from JSON`);
}
