// @openrouter/sdk/esm/models/responsesoutputmessage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 32cb33488ea2
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { OpenAIResponsesRefusalContent$inboundSchema, OpenAIResponsesRefusalContent$outboundSchema, } from "./openairesponsesrefusalcontent.js";
import { ResponseOutputText$inboundSchema, ResponseOutputText$outboundSchema, } from "./responseoutputtext.js";
export const ResponsesOutputMessageRole = {
    Assistant: "assistant",
};
export const ResponsesOutputMessageType = {
    Message: "message",
};
export const ResponsesOutputMessageStatusInProgress = {
    InProgress: "in_progress",
};
export const ResponsesOutputMessageStatusIncomplete = {
    Incomplete: "incomplete",
};
export const ResponsesOutputMessageStatusCompleted = {
    Completed: "completed",
};
/** @internal */
export const ResponsesOutputMessageRole$inboundSchema = z.enum(ResponsesOutputMessageRole);
/** @internal */
export const ResponsesOutputMessageRole$outboundSchema = ResponsesOutputMessageRole$inboundSchema;
/** @internal */
export const ResponsesOutputMessageType$inboundSchema = z.enum(ResponsesOutputMessageType);
/** @internal */
export const ResponsesOutputMessageType$outboundSchema = ResponsesOutputMessageType$inboundSchema;
/** @internal */
export const ResponsesOutputMessageStatusInProgress$inboundSchema = z.enum(ResponsesOutputMessageStatusInProgress);
/** @internal */
export const ResponsesOutputMessageStatusInProgress$outboundSchema = ResponsesOutputMessageStatusInProgress$inboundSchema;
/** @internal */
export const ResponsesOutputMessageStatusIncomplete$inboundSchema = z.enum(ResponsesOutputMessageStatusIncomplete);
/** @internal */
export const ResponsesOutputMessageStatusIncomplete$outboundSchema = ResponsesOutputMessageStatusIncomplete$inboundSchema;
/** @internal */
export const ResponsesOutputMessageStatusCompleted$inboundSchema = z.enum(ResponsesOutputMessageStatusCompleted);
/** @internal */
export const ResponsesOutputMessageStatusCompleted$outboundSchema = ResponsesOutputMessageStatusCompleted$inboundSchema;
/** @internal */
export const ResponsesOutputMessageStatusUnion$inboundSchema = z.union([
    ResponsesOutputMessageStatusCompleted$inboundSchema,
    ResponsesOutputMessageStatusIncomplete$inboundSchema,
    ResponsesOutputMessageStatusInProgress$inboundSchema,
]);
/** @internal */
export const ResponsesOutputMessageStatusUnion$outboundSchema = z.union([
    ResponsesOutputMessageStatusCompleted$outboundSchema,
    ResponsesOutputMessageStatusIncomplete$outboundSchema,
    ResponsesOutputMessageStatusInProgress$outboundSchema,
]);
export function responsesOutputMessageStatusUnionToJSON(responsesOutputMessageStatusUnion) {
    return JSON.stringify(ResponsesOutputMessageStatusUnion$outboundSchema.parse(responsesOutputMessageStatusUnion));
}
export function responsesOutputMessageStatusUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesOutputMessageStatusUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesOutputMessageStatusUnion' from JSON`);
}
/** @internal */
export const ResponsesOutputMessageContent$inboundSchema = z.union([
    ResponseOutputText$inboundSchema,
    OpenAIResponsesRefusalContent$inboundSchema,
]);
/** @internal */
export const ResponsesOutputMessageContent$outboundSchema = z.union([
    ResponseOutputText$outboundSchema,
    OpenAIResponsesRefusalContent$outboundSchema,
]);
export function responsesOutputMessageContentToJSON(responsesOutputMessageContent) {
    return JSON.stringify(ResponsesOutputMessageContent$outboundSchema.parse(responsesOutputMessageContent));
}
export function responsesOutputMessageContentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesOutputMessageContent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesOutputMessageContent' from JSON`);
}
/** @internal */
export const ResponsesOutputMessage$inboundSchema = z.object({
    id: z.string(),
    role: ResponsesOutputMessageRole$inboundSchema,
    type: ResponsesOutputMessageType$inboundSchema,
    status: z.union([
        ResponsesOutputMessageStatusCompleted$inboundSchema,
        ResponsesOutputMessageStatusIncomplete$inboundSchema,
        ResponsesOutputMessageStatusInProgress$inboundSchema,
    ]).optional(),
    content: z.array(z.union([
        ResponseOutputText$inboundSchema,
        OpenAIResponsesRefusalContent$inboundSchema,
    ])),
});
/** @internal */
export const ResponsesOutputMessage$outboundSchema = z.object({
    id: z.string(),
    role: ResponsesOutputMessageRole$outboundSchema,
    type: ResponsesOutputMessageType$outboundSchema,
    status: z.union([
        ResponsesOutputMessageStatusCompleted$outboundSchema,
        ResponsesOutputMessageStatusIncomplete$outboundSchema,
        ResponsesOutputMessageStatusInProgress$outboundSchema,
    ]).optional(),
    content: z.array(z.union([
        ResponseOutputText$outboundSchema,
        OpenAIResponsesRefusalContent$outboundSchema,
    ])),
});
export function responsesOutputMessageToJSON(responsesOutputMessage) {
    return JSON.stringify(ResponsesOutputMessage$outboundSchema.parse(responsesOutputMessage));
}
export function responsesOutputMessageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesOutputMessage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesOutputMessage' from JSON`);
}
