// @openrouter/sdk/esm/models/responseinputaudio.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 259dfb8718b4
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const ResponseInputAudioFormat = {
    Mp3: "mp3",
    Wav: "wav",
};
/** @internal */
export const ResponseInputAudioFormat$inboundSchema = openEnums.inboundSchema(ResponseInputAudioFormat);
/** @internal */
export const ResponseInputAudioFormat$outboundSchema = openEnums.outboundSchema(ResponseInputAudioFormat);
/** @internal */
export const ResponseInputAudioInputAudio$inboundSchema = z.object({
    data: z.string(),
    format: ResponseInputAudioFormat$inboundSchema,
});
/** @internal */
export const ResponseInputAudioInputAudio$outboundSchema = z.object({
    data: z.string(),
    format: ResponseInputAudioFormat$outboundSchema,
});
export function responseInputAudioInputAudioToJSON(responseInputAudioInputAudio) {
    return JSON.stringify(ResponseInputAudioInputAudio$outboundSchema.parse(responseInputAudioInputAudio));
}
export function responseInputAudioInputAudioFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponseInputAudioInputAudio$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponseInputAudioInputAudio' from JSON`);
}
/** @internal */
export const ResponseInputAudio$inboundSchema = z.object({
    type: z.literal("input_audio"),
    input_audio: z.lazy(() => ResponseInputAudioInputAudio$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "input_audio": "inputAudio",
    });
});
/** @internal */
export const ResponseInputAudio$outboundSchema = z.object({
    type: z.literal("input_audio"),
    inputAudio: z.lazy(() => ResponseInputAudioInputAudio$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        inputAudio: "input_audio",
    });
});
export function responseInputAudioToJSON(responseInputAudio) {
    return JSON.stringify(ResponseInputAudio$outboundSchema.parse(responseInputAudio));
}
export function responseInputAudioFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponseInputAudio$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponseInputAudio' from JSON`);
}
