// @openrouter/sdk/esm/models/responseswebsearchuserlocation.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 08377a5f88b1
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
export const ResponsesWebSearchUserLocationType = {
    Approximate: "approximate",
};
/** @internal */
export const ResponsesWebSearchUserLocationType$inboundSchema = z.enum(ResponsesWebSearchUserLocationType);
/** @internal */
export const ResponsesWebSearchUserLocationType$outboundSchema = ResponsesWebSearchUserLocationType$inboundSchema;
/** @internal */
export const ResponsesWebSearchUserLocation$inboundSchema = z.object({
    type: ResponsesWebSearchUserLocationType$inboundSchema.optional(),
    city: z.nullable(z.string()).optional(),
    country: z.nullable(z.string()).optional(),
    region: z.nullable(z.string()).optional(),
    timezone: z.nullable(z.string()).optional(),
});
/** @internal */
export const ResponsesWebSearchUserLocation$outboundSchema = z.object({
    type: ResponsesWebSearchUserLocationType$outboundSchema.optional(),
    city: z.nullable(z.string()).optional(),
    country: z.nullable(z.string()).optional(),
    region: z.nullable(z.string()).optional(),
    timezone: z.nullable(z.string()).optional(),
});
export function responsesWebSearchUserLocationToJSON(responsesWebSearchUserLocation) {
    return JSON.stringify(ResponsesWebSearchUserLocation$outboundSchema.parse(responsesWebSearchUserLocation));
}
export function responsesWebSearchUserLocationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesWebSearchUserLocation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesWebSearchUserLocation' from JSON`);
}
