// @openrouter/sdk/esm/models/responsesoutputitemfilesearchcall.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e5ac80adf28b
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { WebSearchStatus$inboundSchema, WebSearchStatus$outboundSchema, } from "./websearchstatus.js";
export const ResponsesOutputItemFileSearchCallType = {
    FileSearchCall: "file_search_call",
};
/** @internal */
export const ResponsesOutputItemFileSearchCallType$inboundSchema = z.enum(ResponsesOutputItemFileSearchCallType);
/** @internal */
export const ResponsesOutputItemFileSearchCallType$outboundSchema = ResponsesOutputItemFileSearchCallType$inboundSchema;
/** @internal */
export const ResponsesOutputItemFileSearchCall$inboundSchema = z.object({
    type: ResponsesOutputItemFileSearchCallType$inboundSchema,
    id: z.string(),
    queries: z.array(z.string()),
    status: WebSearchStatus$inboundSchema,
});
/** @internal */
export const ResponsesOutputItemFileSearchCall$outboundSchema = z.object({
    type: ResponsesOutputItemFileSearchCallType$outboundSchema,
    id: z.string(),
    queries: z.array(z.string()),
    status: WebSearchStatus$outboundSchema,
});
export function responsesOutputItemFileSearchCallToJSON(responsesOutputItemFileSearchCall) {
    return JSON.stringify(ResponsesOutputItemFileSearchCall$outboundSchema.parse(responsesOutputItemFileSearchCall));
}
export function responsesOutputItemFileSearchCallFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesOutputItemFileSearchCall$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesOutputItemFileSearchCall' from JSON`);
}
