// @openrouter/sdk/esm/models/schema3.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 6f7380bcbcea
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const Schema5 = {
    Unknown: "unknown",
    OpenaiResponsesV1: "openai-responses-v1",
    XaiResponsesV1: "xai-responses-v1",
    AnthropicClaudeV1: "anthropic-claude-v1",
    GoogleGeminiV1: "google-gemini-v1",
};
/** @internal */
export const Schema5$inboundSchema = openEnums
    .inboundSchema(Schema5);
/** @internal */
export const Schema3ReasoningText$inboundSchema = z.object({
    type: z.literal("reasoning.text"),
    text: z.nullable(z.string()).optional(),
    signature: z.nullable(z.string()).optional(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema5$inboundSchema).optional(),
    index: z.number().optional(),
});
export function schema3ReasoningTextFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Schema3ReasoningText$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Schema3ReasoningText' from JSON`);
}
/** @internal */
export const Schema3ReasoningEncrypted$inboundSchema = z.object({
    type: z.literal("reasoning.encrypted"),
    data: z.string(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema5$inboundSchema).optional(),
    index: z.number().optional(),
});
export function schema3ReasoningEncryptedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Schema3ReasoningEncrypted$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Schema3ReasoningEncrypted' from JSON`);
}
/** @internal */
export const Schema3ReasoningSummary$inboundSchema = z.object({
    type: z.literal("reasoning.summary"),
    summary: z.string(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema5$inboundSchema).optional(),
    index: z.number().optional(),
});
export function schema3ReasoningSummaryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Schema3ReasoningSummary$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Schema3ReasoningSummary' from JSON`);
}
/** @internal */
export const Schema3$inboundSchema = z.union([
    z.lazy(() => Schema3ReasoningSummary$inboundSchema),
    z.lazy(() => Schema3ReasoningEncrypted$inboundSchema),
    z.lazy(() => Schema3ReasoningText$inboundSchema),
]);
export function schema3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Schema3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Schema3' from JSON`);
}
