// @openrouter/sdk/esm/models/toolresponsemessage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 1122bbfb530b
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { ChatMessageContentItem$outboundSchema, } from "./chatmessagecontentitem.js";
/** @internal */
export const ToolResponseMessageContent$outboundSchema = z.union([z.string(), z.array(ChatMessageContentItem$outboundSchema)]);
export function toolResponseMessageContentToJSON(toolResponseMessageContent) {
    return JSON.stringify(ToolResponseMessageContent$outboundSchema.parse(toolResponseMessageContent));
}
/** @internal */
export const ToolResponseMessage$outboundSchema = z.object({
    role: z.literal("tool"),
    content: z.union([
        z.string(),
        z.array(ChatMessageContentItem$outboundSchema),
    ]),
    toolCallId: z.string(),
}).transform((v) => {
    return remap$(v, {
        toolCallId: "tool_call_id",
    });
});
export function toolResponseMessageToJSON(toolResponseMessage) {
    return JSON.stringify(ToolResponseMessage$outboundSchema.parse(toolResponseMessage));
}
