// @openrouter/sdk/esm/models/urlcitation.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ccd6ecd62c64
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const URLCitation$inboundSchema = z
    .object({
    type: z.literal("url_citation"),
    url: z.string(),
    title: z.string(),
    start_index: z.number(),
    end_index: z.number(),
}).transform((v) => {
    return remap$(v, {
        "start_index": "startIndex",
        "end_index": "endIndex",
    });
});
/** @internal */
export const URLCitation$outboundSchema = z.object({
    type: z.literal("url_citation"),
    url: z.string(),
    title: z.string(),
    startIndex: z.number(),
    endIndex: z.number(),
}).transform((v) => {
    return remap$(v, {
        startIndex: "start_index",
        endIndex: "end_index",
    });
});
export function urlCitationToJSON(urlCitation) {
    return JSON.stringify(URLCitation$outboundSchema.parse(urlCitation));
}
export function urlCitationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => URLCitation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'URLCitation' from JSON`);
}
