// @openrouter/sdk/esm/models/websearchpreviewtooluserlocation.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: cb1960e09942
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
export const WebSearchPreviewToolUserLocationType = {
    Approximate: "approximate",
};
/** @internal */
export const WebSearchPreviewToolUserLocationType$inboundSchema = z.enum(WebSearchPreviewToolUserLocationType);
/** @internal */
export const WebSearchPreviewToolUserLocationType$outboundSchema = WebSearchPreviewToolUserLocationType$inboundSchema;
/** @internal */
export const WebSearchPreviewToolUserLocation$inboundSchema = z.object({
    type: WebSearchPreviewToolUserLocationType$inboundSchema,
    city: z.nullable(z.string()).optional(),
    country: z.nullable(z.string()).optional(),
    region: z.nullable(z.string()).optional(),
    timezone: z.nullable(z.string()).optional(),
});
/** @internal */
export const WebSearchPreviewToolUserLocation$outboundSchema = z.object({
    type: WebSearchPreviewToolUserLocationType$outboundSchema,
    city: z.nullable(z.string()).optional(),
    country: z.nullable(z.string()).optional(),
    region: z.nullable(z.string()).optional(),
    timezone: z.nullable(z.string()).optional(),
});
export function webSearchPreviewToolUserLocationToJSON(webSearchPreviewToolUserLocation) {
    return JSON.stringify(WebSearchPreviewToolUserLocation$outboundSchema.parse(webSearchPreviewToolUserLocation));
}
export function webSearchPreviewToolUserLocationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebSearchPreviewToolUserLocation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebSearchPreviewToolUserLocation' from JSON`);
}
