// @openrouter/sdk/esm/index.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
// Message format compatibility helpers
export { fromClaudeMessages, toClaudeMessage } from './lib/anthropic-compat.js';
export { hasAsyncFunctions, resolveAsyncFunctions, } from './lib/async-params.js';
export { fromChatMessages, toChatMessage } from './lib/chat-compat.js';
// Claude constants and type guards
export { ClaudeContentBlockType, NonClaudeMessageRole } from './lib/claude-constants.js';
export { isClaudeStyleMessages } from './lib/claude-type-guards.js';
export * from './lib/config.js';
export * as files from './lib/files.js';
export { HTTPClient } from './lib/http.js';
// Next turn params helpers
export { applyNextTurnParamsToRequest, buildNextTurnParamsContext, executeNextTurnParamsFunctions, } from './lib/next-turn-params.js';
// Stop condition helpers
export { finishReasonIs, hasToolCall, isStopConditionMet, maxCost, maxTokensUsed, stepCountIs, } from './lib/stop-conditions.js';
export { extractUnsupportedContent, getUnsupportedContentSummary, hasUnsupportedContent, } from './lib/stream-transformers.js';
// Tool creation helpers
export { tool } from './lib/tool.js';
export { hasExecuteFunction, isGeneratorTool, isRegularExecuteTool, isToolPreliminaryResultEvent, ToolType, } from './lib/tool-types.js';
// Turn context helpers
export { buildTurnContext, normalizeInputToArray } from './lib/turn-context.js';
// Real-time tool event broadcasting
export { ToolEventBroadcaster } from './lib/tool-event-broadcaster.js';
export * from './sdk/sdk.js';
